<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>
<p>An introduction to the <em><a href="https://bioconductor.org/packages/3.8/iSEE">iSEE</a></em> interface</p> • iSEE</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="&lt;p&gt;An introduction to the &lt;em&gt;&lt;a href=" https:>
</head>
<body>
<p>iSEE interface</p>" /&gt;

<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">iSEE</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">1.1.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/basic.html">An introduction to the `r Biocpkg("iSEE")` interface 
</a>
    </li>
    <li>
      <a href="../articles/configure.html">Configuring `r Biocpkg("iSEE")` apps
</a>
    </li>
    <li>
      <a href="../articles/custom.html">Deploying custom panels in the `iSEE` interface</a>
    </li>
    <li>
      <a href="../articles/ecm.html">Describing the `ExperimentColorMap` class</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/csoneson/iSEE">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><link href="basic_files/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet">
<script src="basic_files/font-awesome-5.0.13/js/fontawesome-all.min.js"></script><script src="basic_files/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1></h1>
<p>An introduction to the <em><a href="https://bioconductor.org/packages/3.8/iSEE">iSEE</a></em> interface</p>
                        <h4 class="author">Federico Marini</h4>
            <address class="author_afil">
      Institute of Medical Biostatistics, Epidemiology and Informatics (IMBEI), Mainz<br><a class="author_email" href="mailto:#"></a><a href="mailto:marinif@uni-mainz.de">marinif@uni-mainz.de</a>
      </address>
                              <h4 class="author">Aaron Lun</h4>
            <address class="author_afil">
      Cancer Research UK Cambridge Institute, University of Cambridge<br><a class="author_email" href="mailto:#"></a><a href="mailto:aaron.lun@cruk.cam.ac.uk">aaron.lun@cruk.cam.ac.uk</a>
      </address>
                              <h4 class="author">Charlotte Soneson</h4>
            <address class="author_afil">
      Institute of Molecular Life Sciences, University of Zurich<br><a class="author_email" href="mailto:#"></a><a href="mailto:charlottesoneson@gmail.com">charlottesoneson@gmail.com</a>
      </address>
                              <h4 class="author">Kevin Rue-Albrecht</h4>
            <address class="author_afil">
      Kennedy Institute of Rheumatology, University of Oxford, Headington, Oxford OX3 7FY, UK.<br><a class="author_email" href="mailto:#"></a><a href="mailto:kevin.rue-albrecht@kennedy.ox.ac.uk">kevin.rue-albrecht@kennedy.ox.ac.uk</a>
      </address>
                  
            <h4 class="date">25 September 2018</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/csoneson/iSEE/blob/master/vignettes/basic.Rmd"><code>vignettes/basic.Rmd</code></a></small>
      <div class="hidden name"><code>basic.Rmd</code></div>

    </div>

    
    
<p><strong>Compiled date</strong>: 2018-09-25</p>
<p><strong>Last edited</strong>: 2018-03-08</p>
<p><strong>License</strong>: MIT + file LICENSE</p>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<!-- logo? MOS DEF :) -->
<!-- I'd say TODO -->
<p><em><a href="https://bioconductor.org/packages/3.8/iSEE">iSEE</a></em> is a <a href="http://bioconductor.org">Bioconductor</a> package that provides an interactive Shiny-based graphical user interface for exploring data stored in <code>SummarizedExperiment</code> objects <span class="citation">(Rue-Albrecht et al. 2018)</span>. Instructions to install the package are available <a href="http://bioconductor.org/packages/iSEE/">here</a>. Once installed, the package can be loaded and attached to your current workspace as follows:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(iSEE)</code></pre></div>
<p>If we have a <code>SummarizedExperiment</code> object (or instance of a subclass) named <code>se</code>, we can start the <em><a href="https://bioconductor.org/packages/3.8/iSEE">iSEE</a></em> app by simply calling <code><a href="../reference/iSEE.html">iSEE(se)</a></code>. In this vignette, we will demonstrate this process using the <code>allen</code> single-cell RNA-seq data set from the <em><a href="https://bioconductor.org/packages/3.8/scRNAseq">scRNAseq</a></em> package.</p>
</div>
<div id="setting-up-the-data" class="section level1">
<h1 class="hasAnchor">
<a href="#setting-up-the-data" class="anchor"></a>Setting up the data</h1>
<p>The <code>allen</code> data set contains expression values for 379 cells from the mouse visual cortex <span class="citation">(Tasic et al. 2016)</span>. As provided, the <code>SummarizedExperiment</code> object only contains raw data and not much interesting information. Thus, we need to fill it up for some more exciting visualization.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(scRNAseq)
<span class="kw">data</span>(allen)
<span class="kw">class</span>(allen)
<span class="co">#&gt; [1] "SummarizedExperiment"</span>
<span class="co">#&gt; attr(,"package")</span>
<span class="co">#&gt; [1] "SummarizedExperiment"</span></code></pre></div>
<p>We start by converting the provided <code>SummarizedExperiment</code> object to a <code>SingleCellExperiment</code> object and normalizing the expression values with <em><a href="https://bioconductor.org/packages/3.8/scater">scater</a></em>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(scater)
sce &lt;-<span class="st"> </span><span class="kw">as</span>(allen, <span class="st">"SingleCellExperiment"</span>)
<span class="kw">counts</span>(sce) &lt;-<span class="st"> </span><span class="kw">assay</span>(sce, <span class="st">"tophat_counts"</span>)
sce &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/scater/topics/normalize">normalize</a></span>(sce)</code></pre></div>
<p>Next, we apply PCA and <em>t</em>-SNE to generate two low-dimensional representations of the cells. Note that all computations (e.g., dimension reduction, clustering) must be performed <em>before</em> passing the object to the <code>iSEE</code> function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sce &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/scater/topics/runPCA">runPCA</a></span>(sce)
sce &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/scater/topics/runTSNE">runTSNE</a></span>(sce)
<span class="kw">reducedDimNames</span>(sce)
<span class="co">#&gt; [1] "PCA"  "TSNE"</span></code></pre></div>
<p>The provided cell annotations for this data set are available in <code>colData(sce)</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">colnames</span>(<span class="kw">colData</span>(sce))
<span class="co">#&gt;  [1] "NREADS"                       "NALIGNED"                    </span>
<span class="co">#&gt;  [3] "RALIGN"                       "TOTAL_DUP"                   </span>
<span class="co">#&gt;  [5] "PRIMER"                       "PCT_RIBOSOMAL_BASES"         </span>
<span class="co">#&gt;  [7] "PCT_CODING_BASES"             "PCT_UTR_BASES"               </span>
<span class="co">#&gt;  [9] "PCT_INTRONIC_BASES"           "PCT_INTERGENIC_BASES"        </span>
<span class="co">#&gt; [11] "PCT_MRNA_BASES"               "MEDIAN_CV_COVERAGE"          </span>
<span class="co">#&gt; [13] "MEDIAN_5PRIME_BIAS"           "MEDIAN_3PRIME_BIAS"          </span>
<span class="co">#&gt; [15] "MEDIAN_5PRIME_TO_3PRIME_BIAS" "driver_1_s"                  </span>
<span class="co">#&gt; [17] "dissection_s"                 "Core.Type"                   </span>
<span class="co">#&gt; [19] "Primary.Type"                 "Secondary.Type"              </span>
<span class="co">#&gt; [21] "Animal.ID"                    "passes_qc_checks_s"</span></code></pre></div>
<p>We will also add some per-gene fields to the <code>rowData</code> related to the mean-variance relationship in the data.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">rowData</span>(sce)<span class="op">$</span>mean_log &lt;-<span class="st"> </span><span class="kw">rowMeans</span>(<span class="kw">logcounts</span>(sce))
<span class="kw">rowData</span>(sce)<span class="op">$</span>var_log &lt;-<span class="st"> </span><span class="kw">apply</span>(<span class="kw">logcounts</span>(sce), <span class="dv">1</span>, var)</code></pre></div>
<p>It is important to note that <code>iSEE</code> relies solely<a href="#fn1" class="footnoteRef" id="fnref1"><sup>1</sup></a> on precomputed values in the <code>SummarizedExperiment</code> object. This allows users to visualize any metrics of interest, but also requires them to do so before the initialization of the app.</p>
</div>
<div id="launching-the-interface" class="section level1">
<h1 class="hasAnchor">
<a href="#launching-the-interface" class="anchor"></a>Launching the interface</h1>
<p>To begin the exploration, we create an <code>iSEE</code> app with the <code>SingleCellExperiment</code> object generated above.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">app &lt;-<span class="st"> </span><span class="kw"><a href="../reference/iSEE.html">iSEE</a></span>(sce)</code></pre></div>
<p>We run this using <code>runApp</code> to open the app on our browser.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">shiny<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/shiny/topics/runApp">runApp</a></span>(app)</code></pre></div>
<p>By default, the app starts with a dashboard that contains one panel or table of every type. By opening the collapsible panels named “Data parameters”, “Visual parameters”, and “Selection parameters” under each plot, we can control the content and appearance of each panel.</p>
<p>Now, look in the upper right corner for a <strong>question mark</strong> icon (<i class="fa fa-question-circle"></i>), and click on the button for an introductory tour. This will perform an interactive tour of the app, based on the <em><a href="https://CRAN.R-project.org/package=rintrojs">rintrojs</a></em> package <span class="citation">(Ganz 2016)</span>. During this tour, you will be taken through the different components of the <em><a href="https://bioconductor.org/packages/3.8/iSEE">iSEE</a></em> user interface and learn the basic usage mechanisms by doing: the highlighted elements will be responding to the user’s actions, while the rest of the UI will be shaded. You can move forward and backward along the tour by clicking on the <code>Next</code>/<code>Back</code> buttons, or also using the arrow keys. You can even jump to a particular step by clicking on its circle. To exit the tour, either click on <code>Skip</code>, or simply click outside of the highlighted UI element.</p>
<p>Once you are done generating plots, click on the <strong>wrench</strong> icon (<i class="fa fa-wrench"></i>) in the upper right corner, and click on the button to display R code that you can export and directly re-use in your R session. This will open a modal popup where the R code used to generate the plots is displayed in a <em><a href="https://CRAN.R-project.org/package=shinyAce">shinyAce</a></em>-based text editor. Select parts or all of it to copy-and-paste it into your analysis script/Rmarkdown file.</p>
</div>
<div id="description-of-the-user-interface" class="section level1">
<h1 class="hasAnchor">
<a href="#description-of-the-user-interface" class="anchor"></a>Description of the user interface</h1>
<div id="header" class="section level2">
<h2 class="hasAnchor">
<a href="#header" class="anchor"></a>Header</h2>
<p>The layout of the <em><a href="https://bioconductor.org/packages/3.8/iSEE">iSEE</a></em> user interface is based mostly on the <em><a href="https://CRAN.R-project.org/package=shinydashboard">shinydashboard</a></em> package. The dashboard header contains three dropdown menus. The first is the Diagnostics dropdown menu, which is identified by a <strong>wrench</strong> icon (<i class="fa fa-wrench"></i>) and contains:</p>
<ul>
<li>the “Examine panel chart” functionality, identified by a <strong>chain</strong> icon (<i class="fa fa-chain"></i>). Click on this button to obtain a graph representation of the existing links and point selection among your open plot and table panels (coded with the same colors as in the app). This can be very useful in sessions that include a large number of panels, to visualise the relationship structure between the various panels that send and receive selections of data points.</li>
<li>the “Extract the R code”, functionality (<i class="fa fa-magic"></i>). Once you are done with your live session, you might want to reproduce exactly the plots you generated. Clicking on this button opens a modal popup window, with a <em><a href="https://CRAN.R-project.org/package=shinyAce">shinyAce</a></em>-based text editor, where the code is formatted and displayed with syntax highlighting. You can copy the code to the clipboard by selecting the text (please do include the initial lines and the <code>sessionInfo()</code> commands for best tracking of your environment), and store it in your analysis report/script. Your code can then be further edited to finalize the plots (e.g., for publication).</li>
</ul>
<p>The Documentation dropdown menu is accessible through the <strong>question mark</strong> icon (<i class="fa fa-question-circle"></i>), which contains:</p>
<ul>
<li>the button to start the Quick Tour (<i class="fa fa-hand-o-right"></i>) of <em><a href="https://bioconductor.org/packages/3.8/iSEE">iSEE</a></em>, which allows users to learn the basic usage mechanisms by doing. The highlighted elements will be responding to the user’s actions, while the rest of the UI will be shaded.</li>
<li>the button to “Open the vignette” (<i class="fa fa-book"></i>), which will display the <em><a href="https://bioconductor.org/packages/3.8/iSEE">iSEE</a></em> vignette, either available locally or accessing the current webpage of the package on the Bioconductor project site (if the latter, the vignette will refer to the most current release version, which may differ from the version installed on your system)</li>
</ul>
<p>The Additional Information dropdown menu is accessible through the <strong>information</strong> icon (<i class="fa fa-info"></i>), and contains:</p>
<ul>
<li>the “About this session” button, which reports the output of the <code>sessionInfo()</code> function in a modal popup window (<i class="fa fa-window-maximize"></i>). This is particularly useful for reproducing or reporting the environment, especially in the case of errors or unexpected behaviors.</li>
<li>the “About iSEE” button (<i class="fa fa-heart"></i>) will show up the information on the development team, licensing and citation information for the <em><a href="https://bioconductor.org/packages/3.8/iSEE">iSEE</a></em> package. You can follow the development of the package by checking the GitHub repository (<a href="https://github.com/csoneson/iSEE" class="uri">https://github.com/csoneson/iSEE</a>), where new functionality will be added. Well-considered suggestions in the form of issues and/or pull requests are welcome.</li>
</ul>
</div>
<div id="sidebar" class="section level2">
<h2 class="hasAnchor">
<a href="#sidebar" class="anchor"></a>Sidebar</h2>
<p>The dashboard sidebar of <em><a href="https://bioconductor.org/packages/3.8/iSEE">iSEE</a></em> contains a dropdown menu and a button for creating new plots or tables (referred to as “panels”) in the interface. The maximum number of panels of each type are defined in the <code>iSEE</code> function call that launches the app. Upon clicking these buttons, additional panels of the corresponding type are inserted into the main body of the app. Color-coded tabs in the sidebar contain buttons to change the panel order (<i class="fa fa-arrow-circle-up"></i> and <i class="fa fa-arrow-circle-down"></i>) or to remove panels entirely (<i class="fa fa-trash"></i>). The width and height of each panel can also be adjusted using <i class="fa fa-gear"></i>.</p>
</div>
<div id="body" class="section level2">
<h2 class="hasAnchor">
<a href="#body" class="anchor"></a>Body</h2>
<div id="overview-of-panel-types" class="section level3">
<h3 class="hasAnchor">
<a href="#overview-of-panel-types" class="anchor"></a>Overview of panel types</h3>
<p>The main element in the body of <em><a href="https://bioconductor.org/packages/3.8/iSEE">iSEE</a></em> is the combination of panels, generated (and optionally linked to one another) according to your actions. There are currently six different panel types that can be generated with <em><a href="https://bioconductor.org/packages/3.8/iSEE">iSEE</a></em>:</p>
<ul>
<li>Reduced dimension plots</li>
<li>Column data plots</li>
<li>Feature assay plots</li>
<li>Row statistics tables</li>
<li>Row data plots</li>
<li>Heat maps</li>
</ul>
<p>For each plot panel, three different sets of parameters will be available in collapsible boxes:</p>
<ul>
<li>
<em>Data parameters</em>, to control parameters specific to each type of plot.</li>
<li>
<em>Visual parameters</em>, to specify parameters that will determine the aspect of the plot, in terms of coloring, point features, and more (e.g. legend placement, font size)</li>
<li>
<em>Selection parameters</em> to control the point selection and link relationships to other plots.</li>
</ul>
</div>
<div id="reduced-dimension-plots" class="section level3">
<h3 class="hasAnchor">
<a href="#reduced-dimension-plots" class="anchor"></a>Reduced dimension plots</h3>
<p>If a <code>SingleCellExperiment</code> object is supplied to <code>iSEE</code>, any reduced dimension results are extracted from the <code>reducedDim</code> slot. Examples include low-dimensional embeddings from principal components analysis (PCA) or <em>t</em>-distributed stochastic neighbour embedding (<em>t</em>-SNE) <span class="citation">(Van der Maaten and Hinton 2008)</span>. These results are used to construct two-dimensional <em>reduced dimension plots</em> where each point is a sample, to facilitate efficient exploration of high-dimensional datasets. The <em>Data parameters</em> control the <code>reducedDim</code> slot to be displayed, as well as the two dimensions to plot against each other. Note that <code>iSEE</code> does not computed reduced dimension embeddings; they must be precomputed and available in the object to the <code>iSEE</code> function.</p>
</div>
<div id="coldataplot" class="section level3">
<h3 class="hasAnchor">
<a href="#coldataplot" class="anchor"></a>Column data plots</h3>
<p>A <em>column data plot</em> involves visualizing sample metadata stored in the <code>SummarizedExperiment</code> object. Different fields can be used for the X and Y axes by selecting appropriate values in the plotting parameters. This plot can assume various forms, depending on the nature of the data on the x- and y-axes:</p>
<ul>
<li>If the Y axis is continuous and the X axis is categorical, violin plots are generated (grouped by the X axis factor).</li>
<li>If the Y axis is categorical and the X axis is continuous, horizontal violin plots are generated (grouped by the Y axis factor).</li>
<li>If both are continuous, a scatter plot is generated. This enables the use of contours that are overlaid on top of the plot, check the <code>"Other"</code> box to see the available options.</li>
<li>If both are categorical, a plot of squares is generated where the area of each square is proportional to the number of samples within each combination of factor levels.</li>
</ul>
<p>An X axis setting of “None” is considered to be categorical with a single level.</p>
</div>
<div id="feature-assay-plots" class="section level3">
<h3 class="hasAnchor">
<a href="#feature-assay-plots" class="anchor"></a>Feature assay plots</h3>
<p>A <em>feature assay plot</em> visualizes the assayed values (e.g., gene expression) for a particular feature (e.g., gene) across the samples on the y-axis. This usually results in a (grouped) violin plot, if the X axis is set to <code>"None"</code> or a categorical variable; or a scatter plot, if the X axis is another continuous variable<a href="#fn2" class="footnoteRef" id="fnref2"><sup>2</sup></a>.</p>
<p>Gene selection for the y-axis is achieved by using a <em>linked row statistics table</em> in another panel. Clicking on a row in the table will automatically change the assayed values plotted on the Y axis. Alternatively, the row name can be directly entered as text that corresponds to an entry of <code>rownames(se)</code><a href="#fn3" class="footnoteRef" id="fnref3"><sup>3</sup></a>.</p>
<p>The X axis covariate can also be selected from the plotting parameters. This can be <code>"None"</code>, column data, or the assayed values of another feature (also identified using a linked table or via text). The measurement units are selected as one of the <code>assays(se)</code>, which is applied to both the X and Y axes.</p>
<p>Obviously, any other assayed value for any feature can be visualized in this manner, not limited to the expression of genes. The only requirement for this type of panel is that the observations can be stored as a matrix in the <code>SummarizedExperiment</code> object.</p>
</div>
<div id="row-data-plots" class="section level3">
<h3 class="hasAnchor">
<a href="#row-data-plots" class="anchor"></a>Row data plots</h3>
<p>A <em>row data plot</em> allows the visualization of information stored in the <code>rowData</code> slot of a <code>SummarizedExperiment</code> object. Its behavior mirrors the implementation for the <em>column data plot</em>, and correspondingly this plot can assume various forms depending on whether the data are categorical or continuous.</p>
</div>
<div id="sample-assay-plots" class="section level3">
<h3 class="hasAnchor">
<a href="#sample-assay-plots" class="anchor"></a>Sample assay plots</h3>
<p>A <em>sample assay plot</em> visualizes the assayed values (e.g., gene expression) for a particular sample (e.g., cell) across the features on the Y axis.</p>
<p>This usually results in a (grouped) violin plot, if the X axis is set to <code>"None"</code> or a categorical variable (e.g., gene biotype); or a scatter plot, if the X axis is another continuous variable.</p>
<p>Notably, the X axis covariate can also be set to:</p>
<ul>
<li>a discrete row data covariates (e.g., gene biotype), to stratify the distribution of assayed values</li>
<li>a continuous row data covariate (e.g. count of cells expressing each gene)</li>
<li>another sample, to visualize and compare the assayed values in any two samples.</li>
</ul>
</div>
<div id="row-statistics-tables" class="section level3">
<h3 class="hasAnchor">
<a href="#row-statistics-tables" class="anchor"></a>Row statistics tables</h3>
<p>A <em>row statistics table</em> contains the values of the <code>rowData</code> slot for the <code>SingleCellExperiment</code>/<code>SummarizedExperiment</code> object. If none are available, a column named <code>Present</code> is added and set to <code>TRUE</code> for all available genes, to avoid issues with <code><a href="http://www.rdocumentation.org/packages/DT/topics/datatable">DT::datatable</a></code> and an empty <code>DataFrame</code>. Typically, these tables are used to link to other plots to determine the genes to use for plotting (or coloring). However, they can also be used to retrieve gene-specific annotation on the fly by specifying the <code>annotFun</code> parameter, e.g. using the <code>annotateEntrez</code> or <code>annotateEnsembl</code> functions, provided in <em><a href="https://bioconductor.org/packages/3.8/iSEE">iSEE</a></em>. Alternatively, users can create a customized annotation function; for more details on this, please consult the manual pages <code><a href="../reference/annotateEntrez.html">?annotateEntrez</a></code> and <code><a href="../reference/annotateEnsembl.html">?annotateEnsembl</a></code>.</p>
</div>
<div id="heat-maps" class="section level3">
<h3 class="hasAnchor">
<a href="#heat-maps" class="anchor"></a>Heat maps</h3>
<p>Heat maps provide a compact overview of the data for multiple features in the form of color-coded matrices. These correspond to the <code>assays</code> stored in the <code>SummarizedExperiment</code> object, where features (e.g., genes) are the rows and samples are the columns.</p>
<p>User can select features (rows) to display from the selectize widget (which supports autocompletion), or also via other panels, like row data plots or row statistics tables. The ‘Suggest feature order’ button clusters the rows, and also rearranges the elements in the selectize according to the clustering. It is also possible to choose which assay type is displayed (log-counts being the default choice, if available). The heat map can also be annotated, simply by selecting relevant column data. A zooming functionality is also available, restricted to the Y axis (i.e., allowing closer inspection on the individual features included).</p>
</div>
</div>
</div>
<div id="description-of-isee-functionality" class="section level1">
<h1 class="hasAnchor">
<a href="#description-of-isee-functionality" class="anchor"></a>Description of iSEE functionality</h1>
<div id="coloring-plots-by-sample-attributes" class="section level2">
<h2 class="hasAnchor">
<a href="#coloring-plots-by-sample-attributes" class="anchor"></a>Coloring plots by sample attributes</h2>
<p>Column-based plots are the reduced dimension, feature assay and column data plots, where each point represents a sample. Here, points can be colored in different ways:</p>
<ul>
<li>The default is no color scheme (<code>"None"</code> in the radio button). This results in data points of a constant user-specified color.</li>
<li>Any column of <code>colData(se)</code> can be used. The plot automatically adjusts the scale to use based on whether the chosen column is continuous or categorical.</li>
<li>The assay values of a particular feature in each sample can be used. The feature can be chosen either via a linked row table or selectize input (as described for <em>feature assay plots</em>). Users can also specify the <code>assays</code> from which values are extracted.</li>
<li>The identity of a particular sample can be used, which will be highlighted on the plot in a user-specified color. The sample can be chosen either via a linked column table or via a selectize input.</li>
</ul>
<p>For row-based plots (i.e., the sample assay and row data plots), each point represents a feature. Like the column-based plots, points can be colored by:</p>
<ul>
<li>
<code>"None"</code>, yielding data points of fixed color.</li>
<li>Any column of <code>rowData(se)</code>.</li>
<li>The identity of a particular <em>feature</em>, which is highlighted in the user-specified color.</li>
<li>Assay values for a particular <em>sample</em>.</li>
</ul>
<p>Fine control of the color maps is possible through the <code>ExperimentColorMap</code> class, see <a href="https://bioconductor.org/packages/3.8/iSEE/vignettes/ecm.html">this vignette</a> for more details.</p>
</div>
<div id="controlling-point-aesthetics" class="section level2">
<h2 class="hasAnchor">
<a href="#controlling-point-aesthetics" class="anchor"></a>Controlling point aesthetics</h2>
<p>Points can be set to different shapes according to categorical factors in <code>colData(se)</code> (for column-based plots) or <code>rowData(se)</code> (for row-based plots). This is achieved by checking the <code>"Shape"</code> box to modify the shape-setting options. The size and opacity of the points can be modified via the options available by checking the <code>"Points"</code> box. This may be useful for aesthetically pleasing visualizations when the number of points is very large or small.</p>
<p>It is also possible to downsample points to reduce the time required to generate the plot, via the <code>"Downsample points for speed"</code> checkbox. This uses density-dependent downsampling whereby only the last point for an overlapping set of points is shown. In this manner, we avoid wasting time in plotting many points that would not be visible anyway. Note that this is only effective when the point size is large and the opacity is set to unity.</p>
</div>
<div id="faceting" class="section level2">
<h2 class="hasAnchor">
<a href="#faceting" class="anchor"></a>Faceting</h2>
<p>Each point-based plot can be split into multiple facets using the options in the <code>"Facets"</code> checkbox. Users can facet by row and/or column, using categorical factors in <code>colData(se)</code> (for column-based plots) or <code>rowData(se)</code> (for row-based plots). This provides a convenient way to stratify points in a single plot by multiple factors of interest. Note that data point selection can only occur <em>within</em> each facet, points cannot be selected across facets.</p>
</div>
<div id="select-and-link" class="section level2">
<h2 class="hasAnchor">
<a href="#select-and-link" class="anchor"></a>Selecting data points and linking panels</h2>
<p>To link one plot to another, users can instruct a plotting panel to receive a selection of data points from another (transmitting) plot. This is done using the appropriate field in the “Selection parameters” box. Once this is done, data point selection on the transmitting plot affects the receiving plot in a variety of ways:</p>
<ul>
<li>If the point selection effect is set to <code>"Restrict"</code>, only the subset of points selected in the transmitter are visible in the receiver.</li>
<li>If set to <code>"Color"</code>, the selected subset of points is plotted in the receiver with a user-selected color.</li>
<li>If set to <code>"Transparent"</code>, the selected subset will be drawn with no transparency, while all non-selected points will be plotted with the specified alpha value.</li>
</ul>
<p>It is possible to set up a tree of panels that transmit selections to each other, much like the gating structures used in flow cytometry. This requires all panels acting as both receivers and transmitters to have their effects set to <code>"Restrict"</code> - see <a href="https://github.com/LTLA/iSEE2018/blob/master/tours/cytof/app.R">this mass cytometry tour</a> for an example.</p>
<p>While we have focused on receiving point selections for plots, it is also possible to transmit from plots to tables. This will subset the table so that only the rows corresponding to selected points are visible. Obviously, this means that users should only transmit from row-based plots to row statistics tables, and from column-based plots to column statistics tables.</p>
<p>Custom data panels can receive point selections from both row- and column-based plots, see <a href="https://bioconductor.org/packages/3.8/iSEE/vignettes/custom.html">here</a> for more details.</p>
<p>Note that some points are strictly “end-point” panels and cannot transmit - this includes heat maps, tables and custom data plots.</p>
</div>
<div id="zooming-in-and-out" class="section level2">
<h2 class="hasAnchor">
<a href="#zooming-in-and-out" class="anchor"></a>Zooming in and out</h2>
<p>This is possible by first selecting a region of interest in a plot using the brush (drag and select). Double-clicking on the brushed area then zooms into the selected area. To zoom out to the original plot, simply double-click at any location in the plot.</p>
</div>
</div>
<div id="faq" class="section level1">
<h1 class="hasAnchor">
<a href="#faq" class="anchor"></a>FAQ</h1>
<p><strong>Q: Can you implement a ‘Copy to clipboard’ button in the code editor?</strong></p>
<p>A: This is not necessary, as one can click anywhere in the code editor and instantly select all the code using a keyboard shortcut that depends on your operating system.</p>
<p><strong>Q: When brushing with a transparency effect, it seems that data points in the receiving plot are not subsetted correctly.</strong></p>
<p>A: The subsetting is correct. What you see is an artefact of overplotting: in areas excessively dense in points, transparency ceases to be an effective visual effect.</p>
<p><strong>Q: Brushing on violin or square plots doesn’t seem to select anything.</strong></p>
<p>A: For violin plots, points will be selected only if the brushed area includes the center of the x-tick, i.e., the center of the violin plot. This is intentional as it allows easy selection of all points in complex grouped violin plots. Indeed, the location of a specific point on the x-axis has no meaning. The same logic applies to the square plots, where only the center of each square needs to be selected to obtain all the points in the square.</p>
<p><strong>Q: I’d like to try <em><a href="https://bioconductor.org/packages/3.8/iSEE">iSEE</a></em> but I can’t install it/I just want a quick peek. Is there something you can do?</strong></p>
<p>A: We set up an instance of iSEE running on the <code>allen</code> dataset at this address: <a href="http://shiny.imbei.uni-mainz.de:3838/iSEE" class="uri">http://shiny.imbei.uni-mainz.de:3838/iSEE</a>. Please keep in mind this is only for trial purposes, yet it can show a quick way of how you or your system administrator can setup <em><a href="https://bioconductor.org/packages/3.8/iSEE">iSEE</a></em> for analyzing your <code>SummarizedExperiment</code>/<code>SingleCellExperiment</code> precomputed object.</p>
</div>
<div id="additional-information" class="section level1">
<h1 class="hasAnchor">
<a href="#additional-information" class="anchor"></a>Additional information</h1>
<p>Bug reports can be posted on the <a href="https://support.bioconductor.org">Bioconductor support site</a> or raised as issues in the <em><a href="https://github.com/csoneson/iSEE">iSEE</a></em> GitHub repository. The GitHub repository also contains the development version of the package, where new functionality will be added in the future. The authors appreciate well-considered suggestions for improvements or new features, or even better, pull requests.</p>
<p>If you use <em><a href="https://bioconductor.org/packages/3.8/iSEE">iSEE</a></em> for your analysis, please cite it as shown below:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">citation</span>(<span class="st">"iSEE"</span>)
<span class="co">#&gt; </span>
<span class="co">#&gt; Rue-Albrecht K, Marini F, Soneson C, Lun ATL (2018). "iSEE:</span>
<span class="co">#&gt; Interactive SummarizedExperiment Explorer." _F1000Research_, *7*,</span>
<span class="co">#&gt; 741. doi: 10.12688/f1000research.14966.1 (URL:</span>
<span class="co">#&gt; http://doi.org/10.12688/f1000research.14966.1).</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; A BibTeX entry for LaTeX users is</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   @Article{,</span>
<span class="co">#&gt;     title = {iSEE: Interactive SummarizedExperiment Explorer},</span>
<span class="co">#&gt;     author = {Kevin Rue-Albrecht and Federico Marini and Charlotte Soneson and Aaron T. L. Lun},</span>
<span class="co">#&gt;     publisher = {F1000 Research, Ltd.},</span>
<span class="co">#&gt;     journal = {F1000Research},</span>
<span class="co">#&gt;     year = {2018},</span>
<span class="co">#&gt;     month = {Jun},</span>
<span class="co">#&gt;     volume = {7},</span>
<span class="co">#&gt;     pages = {741},</span>
<span class="co">#&gt;     doi = {10.12688/f1000research.14966.1},</span>
<span class="co">#&gt;   }</span></code></pre></div>
</div>
<div id="session-info" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()
<span class="co">#&gt; R version 3.5.0 (2018-04-23)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin15.6.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Sierra 10.12.6</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] parallel  stats4    stats     graphics  grDevices utils     datasets </span>
<span class="co">#&gt; [8] methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] scater_1.9.20               ggplot2_3.0.0              </span>
<span class="co">#&gt;  [3] scRNAseq_1.7.0              iSEE_1.1.9                 </span>
<span class="co">#&gt;  [5] SingleCellExperiment_1.3.10 SummarizedExperiment_1.11.6</span>
<span class="co">#&gt;  [7] DelayedArray_0.7.42         BiocParallel_1.15.12       </span>
<span class="co">#&gt;  [9] matrixStats_0.54.0          Biobase_2.41.2             </span>
<span class="co">#&gt; [11] GenomicRanges_1.33.14       GenomeInfoDb_1.17.1        </span>
<span class="co">#&gt; [13] IRanges_2.15.18             S4Vectors_0.19.19          </span>
<span class="co">#&gt; [15] BiocGenerics_0.27.1         BiocStyle_2.9.6            </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] nlme_3.1-137             bitops_1.0-6            </span>
<span class="co">#&gt;  [3] fs_1.2.6                 bit64_0.9-7             </span>
<span class="co">#&gt;  [5] rprojroot_1.3-2          tools_3.5.0             </span>
<span class="co">#&gt;  [7] backports_1.1.2          R6_2.2.2                </span>
<span class="co">#&gt;  [9] DT_0.4                   HDF5Array_1.9.19        </span>
<span class="co">#&gt; [11] vipor_0.4.5              DBI_1.0.0               </span>
<span class="co">#&gt; [13] lazyeval_0.2.1           mgcv_1.8-24             </span>
<span class="co">#&gt; [15] colorspace_1.3-2         withr_2.1.2             </span>
<span class="co">#&gt; [17] gridExtra_2.3            tidyselect_0.2.4        </span>
<span class="co">#&gt; [19] bit_1.1-14               compiler_3.5.0          </span>
<span class="co">#&gt; [21] shinyjs_1.0              xml2_1.2.0              </span>
<span class="co">#&gt; [23] desc_1.2.0               colourpicker_1.0        </span>
<span class="co">#&gt; [25] bookdown_0.7             scales_1.0.0            </span>
<span class="co">#&gt; [27] pkgdown_1.1.0            commonmark_1.5          </span>
<span class="co">#&gt; [29] stringr_1.3.1            digest_0.6.17           </span>
<span class="co">#&gt; [31] rmarkdown_1.10           rentrez_1.2.1           </span>
<span class="co">#&gt; [33] XVector_0.21.3           pkgconfig_2.0.2         </span>
<span class="co">#&gt; [35] htmltools_0.3.6          htmlwidgets_1.2         </span>
<span class="co">#&gt; [37] rlang_0.2.2              rstudioapi_0.7          </span>
<span class="co">#&gt; [39] RSQLite_2.1.1            DelayedMatrixStats_1.3.9</span>
<span class="co">#&gt; [41] shiny_1.1.0              bindr_0.1.1             </span>
<span class="co">#&gt; [43] jsonlite_1.5             dplyr_0.7.6             </span>
<span class="co">#&gt; [45] RCurl_1.95-4.11          magrittr_1.5            </span>
<span class="co">#&gt; [47] GenomeInfoDbData_1.1.0   Matrix_1.2-14           </span>
<span class="co">#&gt; [49] ggbeeswarm_0.6.0         Rhdf5lib_1.3.3          </span>
<span class="co">#&gt; [51] Rcpp_0.12.18             munsell_0.5.0           </span>
<span class="co">#&gt; [53] viridis_0.5.1            rintrojs_0.2.0          </span>
<span class="co">#&gt; [55] stringi_1.2.4            yaml_2.2.0              </span>
<span class="co">#&gt; [57] MASS_7.3-50              zlibbioc_1.27.0         </span>
<span class="co">#&gt; [59] Rtsne_0.13               rhdf5_2.25.11           </span>
<span class="co">#&gt; [61] plyr_1.8.4               grid_3.5.0              </span>
<span class="co">#&gt; [63] blob_1.1.1               promises_1.0.1          </span>
<span class="co">#&gt; [65] shinydashboard_0.7.0     crayon_1.3.4            </span>
<span class="co">#&gt; [67] miniUI_0.1.1.1           lattice_0.20-35         </span>
<span class="co">#&gt; [69] cowplot_0.9.3            knitr_1.20              </span>
<span class="co">#&gt; [71] pillar_1.3.0             igraph_1.2.2            </span>
<span class="co">#&gt; [73] reshape2_1.4.3           XML_3.98-1.16           </span>
<span class="co">#&gt; [75] glue_1.3.0               evaluate_0.11           </span>
<span class="co">#&gt; [77] BiocManager_1.30.2       httpuv_1.4.5            </span>
<span class="co">#&gt; [79] gtable_0.2.0             purrr_0.2.5             </span>
<span class="co">#&gt; [81] assertthat_0.2.0         xfun_0.3                </span>
<span class="co">#&gt; [83] mime_0.5                 xtable_1.8-3            </span>
<span class="co">#&gt; [85] roxygen2_6.1.0           later_0.7.5             </span>
<span class="co">#&gt; [87] viridisLite_0.3.0        tibble_1.4.2            </span>
<span class="co">#&gt; [89] beeswarm_0.2.3           AnnotationDbi_1.43.1    </span>
<span class="co">#&gt; [91] memoise_1.1.0            bindrcpp_0.2.2          </span>
<span class="co">#&gt; [93] shinyAce_0.3.1</span>
<span class="co"># devtools::session_info()</span></code></pre></div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-ganz2016rintrojs">
<p>Ganz, C. 2016. “rintrojs: A Wrapper for the Intro.js Library.” <em>Journal of Open Source Software</em> 1 (6). The Open Journal. <a href="http://dx.doi.org/10.21105/joss.00063" class="uri">http://dx.doi.org/10.21105/joss.00063</a>.</p>
</div>
<div id="ref-kra2018iSEE">
<p>Rue-Albrecht, K., F. Marini, C. Soneson, and A. T. L. Lun. 2018. “ISEE: Interactive Summarizedexperiment Explorer.” <em>F1000Research</em> 7 (June): 741.</p>
</div>
<div id="ref-tasic2016adult">
<p>Tasic, B., V. Menon, T. N. Nguyen, T. K. Kim, T. Jarsky, Z. Yao, B. Levi, et al. 2016. “Adult mouse cortical cell taxonomy revealed by single cell transcriptomics.” <em>Nat. Neurosci.</em> 19 (2): 335–46.</p>
</div>
<div id="ref-van2008visualizing">
<p>Van der Maaten, L., and G. Hinton. 2008. “Visualizing Data Using T-SNE.” <em>J. Mach. Learn. Res.</em> 9 (2579-2605): 85.</p>
</div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>Except when dealing with <a href="https://bioconductor.org/packages/3.8/iSEE/vignettes/custom.html">custom panels</a>.<a href="#fnref1">↩</a></p></li>
<li id="fn2"><p>That said, if there are categorical values for the assayed values, these will be handled as described in the column data plots.<a href="#fnref2">↩</a></p></li>
<li id="fn3"><p>This is not effective if <code>se</code> does not contain row names.<a href="#fnref3">↩</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li><a href="#setting-up-the-data">Setting up the data</a></li>
      <li><a href="#launching-the-interface">Launching the interface</a></li>
      <li>
<a href="#description-of-the-user-interface">Description of the user interface</a><ul class="nav nav-pills nav-stacked">
<li><a href="#header">Header</a></li>
      <li><a href="#sidebar">Sidebar</a></li>
      <li><a href="#body">Body</a></li>
      </ul>
</li>
      <li>
<a href="#description-of-isee-functionality">Description of iSEE functionality</a><ul class="nav nav-pills nav-stacked">
<li><a href="#coloring-plots-by-sample-attributes">Coloring plots by sample attributes</a></li>
      <li><a href="#controlling-point-aesthetics">Controlling point aesthetics</a></li>
      <li><a href="#faceting">Faceting</a></li>
      <li><a href="#select-and-link">Selecting data points and linking panels</a></li>
      <li><a href="#zooming-in-and-out">Zooming in and out</a></li>
      </ul>
</li>
      <li><a href="#faq">FAQ</a></li>
      <li><a href="#additional-information">Additional information</a></li>
      <li><a href="#session-info">Session Info</a></li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by <a href="link_to_charlotte">Charlotte Soneson</a>, <a href="link_to_aaron">Aaron Lun</a>, <a href="https://federicomarini.github.io">Federico Marini</a>, <a href="link_to_kevin">Kevin Rue-Albrecht</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  
</body>
</html>
