<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>
<p>Configuring <em><a href="https://bioconductor.org/packages/3.8/iSEE">iSEE</a></em> apps</p> • iSEE</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="&lt;p&gt;Configuring &lt;em&gt;&lt;a href=" https:>
</head>
<body>
<p>iSEE apps</p>" /&gt;

<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">iSEE</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">1.1.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/basic.html">An introduction to the `r Biocpkg("iSEE")` interface 
</a>
    </li>
    <li>
      <a href="../articles/configure.html">Configuring `r Biocpkg("iSEE")` apps
</a>
    </li>
    <li>
      <a href="../articles/custom.html">Deploying custom panels in the `iSEE` interface</a>
    </li>
    <li>
      <a href="../articles/ecm.html">Describing the `ExperimentColorMap` class</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/csoneson/iSEE">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><link href="configure_files/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet">
<script src="configure_files/font-awesome-5.0.13/js/fontawesome-all.min.js"></script><script src="configure_files/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1></h1>
<p>Configuring <em><a href="https://bioconductor.org/packages/3.8/iSEE">iSEE</a></em> apps</p>
                        <h4 class="author">Federico Marini</h4>
            <address class="author_afil">
      Institute of Medical Biostatistics, Epidemiology and Informatics (IMBEI), Mainz<br><a class="author_email" href="mailto:#"></a><a href="mailto:marinif@uni-mainz.de">marinif@uni-mainz.de</a>
      </address>
                              <h4 class="author">Aaron Lun</h4>
            <address class="author_afil">
      Cancer Research UK Cambridge Institute, University of Cambridge<br><a class="author_email" href="mailto:#"></a><a href="mailto:aaron.lun@cruk.cam.ac.uk">aaron.lun@cruk.cam.ac.uk</a>
      </address>
                              <h4 class="author">Charlotte Soneson</h4>
            <address class="author_afil">
      Institute of Molecular Life Sciences, University of Zurich<br><a class="author_email" href="mailto:#"></a><a href="mailto:charlottesoneson@gmail.com">charlottesoneson@gmail.com</a>
      </address>
                              <h4 class="author">Kevin Rue-Albrecht</h4>
            <address class="author_afil">
      Kennedy Institute of Rheumatology, University of Oxford, Headington, Oxford OX3 7FY, UK.<br><a class="author_email" href="mailto:#"></a><a href="mailto:kevin.rue-albrecht@kennedy.ox.ac.uk">kevin.rue-albrecht@kennedy.ox.ac.uk</a>
      </address>
                  
            <h4 class="date">25 September 2018</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/csoneson/iSEE/blob/master/vignettes/configure.Rmd"><code>vignettes/configure.Rmd</code></a></small>
      <div class="hidden name"><code>configure.Rmd</code></div>

    </div>

    
    
<p><strong>Compiled date</strong>: 2018-09-25</p>
<p><strong>Last edited</strong>: 2018-03-08</p>
<p><strong>License</strong>: MIT + file LICENSE</p>
<div id="changing-the-default-start-configuration" class="section level1">
<h1 class="hasAnchor">
<a href="#changing-the-default-start-configuration" class="anchor"></a>Changing the default start configuration</h1>
<p>The default start configuration with one plot of each type may not always be the most appropriate. <code>iSEE</code> allows the user to programmatically modify the initial settings <span class="citation">(Rue-Albrecht et al. 2018)</span>, avoiding the need to click through the choices to obtain the desired panel setup. Almost every aspect of the initial app state can be customized, down to whether or not the parameter boxes are open or closed!</p>
<p>To demonstrate this feature, let’s assume that we are only interested in <em>feature assay plot</em> panels. The default set of panels can be changed via the <code>initialPanels</code> argument of the <code>iSEE</code> function call. Given a <code>SingleCellExperiment</code> or <code>SummarizedExperiment</code> object named <code>sce</code><a href="#fn1" class="footnoteRef" id="fnref1"><sup>1</sup></a>, the following code opens an app with two adjacent feature assay plots. Note that each panel is set to occupy half the width of the application window, which is set to 12 units by the <em><a href="https://CRAN.R-project.org/package=shiny">shiny</a></em> package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(iSEE)
init &lt;-<span class="st"> </span><span class="kw">DataFrame</span>(
    <span class="dt">Name =</span> <span class="kw">c</span>(<span class="st">"Feature assay plot 1"</span>, <span class="st">"Feature assay plot 2"</span>),
    <span class="dt">Width =</span> <span class="kw">c</span>(<span class="dv">6</span>, <span class="dv">6</span>)
)
app &lt;-<span class="st"> </span><span class="kw"><a href="../reference/iSEE.html">iSEE</a></span>(sce, <span class="dt">initialPanels =</span> init)</code></pre></div>
<p>The genes to show on the Y-axis in the two plots can be specified via the <code>featAssayArgs</code> argument to <code>iSEE</code>. This is a <code>DataFrame</code>, specifying the initial parameters for each plot. In this case, we want to modify the <code>YAxis</code> and <code>YAxisFeatName</code> defaults for the two plots:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fexArg &lt;-<span class="st"> </span><span class="kw"><a href="../reference/featAssayPlotDefaults.html">featAssayPlotDefaults</a></span>(sce, <span class="dv">2</span>)
fexArg<span class="op">$</span>YAxisFeatName &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"0610009L18Rik"</span>, <span class="st">"0610009B22Rik"</span>)
app &lt;-<span class="st"> </span><span class="kw"><a href="../reference/iSEE.html">iSEE</a></span>(sce, <span class="dt">initialPanels =</span> init, <span class="dt">featAssayArgs =</span> fexArg)</code></pre></div>
<p>This will open the app with two feature assay plots, showing the selected genes. Of course, from this starting point, all the interactive functionality is still available, and new panels can be added, modified and linked via point selection.</p>
</div>
<div id="data-parameters" class="section level1">
<h1 class="hasAnchor">
<a href="#data-parameters" class="anchor"></a>Data parameters</h1>
<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<p>The <em>data parameters</em> control the source of the information represented on the X-axis and Y-axis of each plot. Those parameters are accessible at runtime in the eponymous collapsible box.</p>
<p>We refer users to the individual help page of each panel type listed <a href="#further">below</a> to learn more about the choices of X-axis variables for each type of panel.</p>
<p>From a running <code>iSEE</code> application, you can also display all the R code that is required to set up the current configuration by clicking on <code>Display panel settings</code> under the wrench icon in the top-right corner.</p>
</div>
<div id="setting-the-y-axis" class="section level2">
<h2 class="hasAnchor">
<a href="#setting-the-y-axis" class="anchor"></a>Setting the Y-axis</h2>
<p>The nature of the Y-axis is defined by the type of panel. For instance, <em>column data plot</em> panels require the name of a column in the <code>colData(sce)</code>. Users can preconfigure the Y-axis of individual <em>column data plot</em> panels as follows:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cdArgs &lt;-<span class="st"> </span><span class="kw"><a href="../reference/colDataPlotDefaults.html">colDataPlotDefaults</a></span>(sce, <span class="dv">2</span>)
cdArgs<span class="op">$</span>DataBoxOpen &lt;-<span class="st"> </span><span class="ot">TRUE</span>
cdArgs<span class="op">$</span>YAxis &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"NREADS"</span>, <span class="st">"TOTAL_DUP"</span>)
init &lt;-<span class="st"> </span><span class="kw">DataFrame</span>(
    <span class="dt">Name =</span> <span class="kw">c</span>(<span class="st">"Column data plot 1"</span>, <span class="st">"Column data plot 2"</span>),
    <span class="dt">Width =</span> <span class="kw">c</span>(<span class="dv">6</span>, <span class="dv">6</span>)
)
app &lt;-<span class="st"> </span><span class="kw"><a href="../reference/iSEE.html">iSEE</a></span>(sce, <span class="dt">initialPanels =</span> init, <span class="dt">colDataArgs =</span> cdArgs)</code></pre></div>
</div>
<div id="xaxis" class="section level2">
<h2 class="hasAnchor">
<a href="#xaxis" class="anchor"></a>Setting the X-axis</h2>
<p>The X-axis can be set to different types of variables, which generally requires setting at least two columns in the <code>DataFrame</code> of arguments. The type of variable is generally set in the <code>XAxis</code> column, while the name of the variable is stored in a different column for each type of variable. At runtime, this allows the app to remember the last selected variable of each type.</p>
<p>For instance, the <code>XAxis</code> column of <em>feature assay plot</em> arguments can have up to four different values:</p>
<ol style="list-style-type: decimal">
<li>
<code>"None"</code>: does not require any addition input (draws a single violin for all features)</li>
<li>
<code>"Column data"</code>: requires <code>XAxisColData</code> to be set to a column name in the <code>colData(sce)</code>
</li>
<li>
<code>"Feature name"</code>: requires either
<ol style="list-style-type: lower-alpha">
<li>
<code>XAxisFeatName</code> to be set to a feature name (or positional index) in <code>rownames(sce)</code>
</li>
<li>
<code>XAxisRowTable</code> to be set to the name of a <em>Row statistics table</em> panel with an active selection set in its own <code>Selected</code> column</li>
</ol>
</li>
</ol>
<p>Each of these scenarios is demonstrated below:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fexArg &lt;-<span class="st"> </span><span class="kw"><a href="../reference/featAssayPlotDefaults.html">featAssayPlotDefaults</a></span>(sce, <span class="dv">4</span>)
fexArg<span class="op">$</span>DataBoxOpen &lt;-<span class="st"> </span><span class="ot">TRUE</span>

<span class="co"># Example 1</span>
fexArg[<span class="dv">1</span>, <span class="st">"XAxis"</span>] &lt;-<span class="st"> "None"</span>

<span class="co"># Example 2</span>
fexArg[<span class="dv">2</span>, <span class="st">"XAxis"</span>] &lt;-<span class="st"> "Column data"</span>
fexArg[<span class="dv">2</span>, <span class="st">"XAxisColData"</span>] &lt;-<span class="st"> "Core.Type"</span>

<span class="co"># Example 3a</span>
fexArg[<span class="dv">3</span>, <span class="st">"XAxis"</span>] &lt;-<span class="st"> "Feature name"</span>
fexArg[<span class="dv">3</span>, <span class="st">"XAxisFeatName"</span>] &lt;-<span class="st"> "Zyx"</span>

<span class="co"># Example 3b (also requires a row statistic table)</span>
fexArg[<span class="dv">4</span>, <span class="st">"XAxis"</span>] &lt;-<span class="st"> "Feature name"</span>
fexArg[<span class="dv">4</span>, <span class="st">"XAxisRowTable"</span>] &lt;-<span class="st"> "Row statistics table 1"</span>
<span class="kw">rowData</span>(sce)[, <span class="st">"gene_id"</span>] &lt;-<span class="st"> </span><span class="kw">rownames</span>(sce)
rsArg &lt;-<span class="st"> </span><span class="kw"><a href="../reference/rowStatTableDefaults.html">rowStatTableDefaults</a></span>(sce, <span class="dv">1</span>)
rsArg<span class="op">$</span>Selected &lt;-<span class="st"> "Ints2"</span>
rsArg<span class="op">$</span>SearchColumns &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="dt">gene_id=</span><span class="st">"Ints"</span>)

<span class="co"># Initialisation</span>
init &lt;-<span class="st"> </span><span class="kw">DataFrame</span>(
    <span class="dt">Name =</span> <span class="kw">c</span>(
        <span class="kw">paste</span>(<span class="st">"Feature assay plot"</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">4</span>),
        <span class="st">"Row statistics table 1"</span>),
    <span class="dt">Width =</span> <span class="kw">c</span>(<span class="kw">rep</span>(<span class="dv">6</span>, <span class="dv">4</span>), <span class="dv">12</span>)
)
app &lt;-<span class="st"> </span><span class="kw"><a href="../reference/iSEE.html">iSEE</a></span>(sce, <span class="dt">initialPanels =</span> init, <span class="dt">featAssayArgs =</span> fexArg, <span class="dt">rowStatArgs =</span> rsArg)</code></pre></div>
<p>Note how <em>Example 3b</em> requires an active <em>row statistics table</em> as a source of selection. To facilitate visualisation, we added the features identifiers as the <code>gene_id</code> column in <code>rowData(sce)</code>, we preselected the feature <code>"Ints2"</code>, and we prefiltered the table using the pattern <code>"Ints"</code> on the <code>gene_id</code> column to show this active selection.</p>
</div>
<div id="configuring-the-type-of-dimensionality-reduction" class="section level2">
<h2 class="hasAnchor">
<a href="#configuring-the-type-of-dimensionality-reduction" class="anchor"></a>Configuring the type of dimensionality reduction</h2>
<p>In the case of reduced dimension plots, <em>data parameters</em> include the name of the reduced dimension slot from which to fetch coordinates. This information is stored in the <code>Type</code> column:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">rdArgs &lt;-<span class="st"> </span><span class="kw"><a href="../reference/redDimPlotDefaults.html">redDimPlotDefaults</a></span>(sce, <span class="dv">1</span>)
rdArgs<span class="op">$</span>DataBoxOpen &lt;-<span class="st"> </span><span class="ot">TRUE</span>
rdArgs<span class="op">$</span>Type &lt;-<span class="st"> "TSNE"</span>
rdArgs<span class="op">$</span>XAxis &lt;-<span class="st"> </span><span class="dv">2</span>
rdArgs<span class="op">$</span>YAxis &lt;-<span class="st"> </span><span class="dv">1</span>
init &lt;-<span class="st"> </span><span class="kw">DataFrame</span>(
    <span class="dt">Name =</span> <span class="kw">c</span>(<span class="st">"Reduced dimension plot 1"</span>),
    <span class="dt">Width =</span> <span class="kw">c</span>(<span class="dv">6</span>)
)
app &lt;-<span class="st"> </span><span class="kw"><a href="../reference/iSEE.html">iSEE</a></span>(sce, <span class="dt">initialPanels =</span> init, <span class="dt">redDimArgs =</span> rdArgs)</code></pre></div>
<p>We refer users to the <code><a href="../reference/redDimPlotDefaults.html">?redDimPlotDefaults</a></code> help page to learn more about the choices of default parameters for <em>reduced dimension plot</em> panels.</p>
</div>
<div id="assay" class="section level2">
<h2 class="hasAnchor">
<a href="#assay" class="anchor"></a>Configuring the type of assay data</h2>
<p>For axes linked to an assay, such as the Y-axis of <em>feature assay plot</em> panels, the assay to display can be set through the <code>Assay</code> column of the arguments:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fexArg &lt;-<span class="st"> </span><span class="kw"><a href="../reference/featAssayPlotDefaults.html">featAssayPlotDefaults</a></span>(sce, <span class="dv">2</span>)
fexArg<span class="op">$</span>DataBoxOpen &lt;-<span class="st"> </span><span class="ot">TRUE</span>
fexArg<span class="op">$</span>Assay &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"rsem_counts"</span>, <span class="st">"tophat_counts"</span>)
init &lt;-<span class="st"> </span><span class="kw">DataFrame</span>(
    <span class="dt">Name =</span> <span class="kw">paste</span>(<span class="st">"Feature assay plot"</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">2</span>),
    <span class="dt">Width =</span> <span class="kw">c</span>(<span class="dv">6</span>, <span class="dv">6</span>)
)
app &lt;-<span class="st"> </span><span class="kw"><a href="../reference/iSEE.html">iSEE</a></span>(sce, <span class="dt">initialPanels =</span> init, <span class="dt">featAssayArgs =</span> fexArg)</code></pre></div>
</div>
</div>
<div id="visual-parameters" class="section level1">
<h1 class="hasAnchor">
<a href="#visual-parameters" class="anchor"></a>Visual parameters</h1>
<div id="overview-1" class="section level2">
<h2 class="hasAnchor">
<a href="#overview-1" class="anchor"></a>Overview</h2>
<p>The <em>visual parameters</em> control the appearance of data points. Those parameters include: color, shape, size, opacity, facet, as well as font size and legend position. Some visual parameters can be associated to variables and controlled through <em><a href="https://CRAN.R-project.org/package=ggplot2">ggplot2</a></em> aesthetics, while others can be set to constant user-defined values. All those parameters are accessible at runtime in the eponymous collapsible box.</p>
<p>We refer users to the <code><a href="../reference/pointDefaults.html">?"iSEE point parameters"</a></code> help page to learn more about the visual parameters variables configurable for each type of panel; and to the <code><a href="../reference/selectDefaults.html">?"iSEE selection parameters"</a></code> help page to learn more about the choices of parameters that control the appearance of point selections in receiver plot panels.</p>
</div>
<div id="configuring-default-visual-parameters" class="section level2">
<h2 class="hasAnchor">
<a href="#configuring-default-visual-parameters" class="anchor"></a>Configuring default visual parameters</h2>
<p>Certain visual parameters can be set to a constant user-defined value. Those include: color, transparency (i.e., alpha), downsampling resolution, as well as text font size and legend position.</p>
<p>For instance, the default color of data points in <em>column data plot</em> panels can be set to a value different than the default <code>"black"</code> through the <code>ColorByDefaultColor</code> column, while the default transparency value is controlled through the <code>PointAlpha</code> column. Here, we alter several default visual parameters in the second panel:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cdArgs &lt;-<span class="st"> </span><span class="kw"><a href="../reference/colDataPlotDefaults.html">colDataPlotDefaults</a></span>(sce, <span class="dv">2</span>)
cdArgs<span class="op">$</span>VisualBoxOpen &lt;-<span class="st"> </span><span class="ot">TRUE</span>
cdArgs<span class="op">$</span>VisualChoices &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="kw">c</span>(<span class="st">"Color"</span>, <span class="st">"Points"</span>, <span class="st">"Other"</span>), <span class="kw">c</span>(<span class="st">"Color"</span>, <span class="st">"Points"</span>, <span class="st">"Other"</span>))
cdArgs<span class="op">$</span>ColorByDefaultColor[<span class="dv">2</span>] &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"chocolate3"</span>)
cdArgs<span class="op">$</span>PointAlpha[<span class="dv">2</span>] &lt;-<span class="st"> </span><span class="fl">0.2</span>
cdArgs<span class="op">$</span>PointSize[<span class="dv">2</span>] &lt;-<span class="st"> </span><span class="dv">2</span>
cdArgs<span class="op">$</span>Downsample[<span class="dv">2</span>] &lt;-<span class="st"> </span><span class="ot">TRUE</span>
cdArgs<span class="op">$</span>SampleRes[<span class="dv">2</span>] &lt;-<span class="st"> </span><span class="dv">150</span>
cdArgs<span class="op">$</span>FontSize[<span class="dv">2</span>] &lt;-<span class="st"> </span><span class="dv">2</span>

init &lt;-<span class="st"> </span><span class="kw">DataFrame</span>(
    <span class="dt">Name =</span> <span class="kw">c</span>(<span class="st">"Column data plot 1"</span>, <span class="st">"Column data plot 2"</span>),
    <span class="dt">Width =</span> <span class="kw">c</span>(<span class="dv">6</span>, <span class="dv">6</span>)
)
app &lt;-<span class="st"> </span><span class="kw"><a href="../reference/iSEE.html">iSEE</a></span>(sce, <span class="dt">initialPanels =</span> init, <span class="dt">colDataArgs =</span> cdArgs)</code></pre></div>
<p>Note that for this demonstration, we facilitate visualization of the preconfigured arguments by setting <code>VisualChoices</code> to display both the <code>"Color"</code> and <code>"Shape"</code> UI panels.</p>
</div>
<div id="linking-point-aesthetics-to-variables" class="section level2">
<h2 class="hasAnchor">
<a href="#linking-point-aesthetics-to-variables" class="anchor"></a>Linking point aesthetics to variables</h2>
<p>The color and point of data points can be linked to variables in a manner similar to the X-axis parameters demonstrated <a href="#xaxis">above</a>.</p>
<p>For instance, the color of data points in <em>column data plot</em> panels can be set to a variable in <code>colData(sce)</code> by setting the <code>ColorBy</code> value to <code>"Column data"</code>, and the <code>ColorByColData</code> value to the desired coloumn name:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cdArgs &lt;-<span class="st"> </span><span class="kw"><a href="../reference/colDataPlotDefaults.html">colDataPlotDefaults</a></span>(sce, <span class="dv">2</span>)
cdArgs<span class="op">$</span>VisualBoxOpen &lt;-<span class="st"> </span><span class="ot">TRUE</span>
cdArgs<span class="op">$</span>VisualChoices &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="kw">c</span>(<span class="st">"Color"</span>, <span class="st">"Shape"</span>), <span class="kw">c</span>(<span class="st">"Color"</span>, <span class="st">"Shape"</span>))
cdArgs<span class="op">$</span>ColorBy &lt;-<span class="st"> "Column data"</span>
cdArgs<span class="op">$</span>ShapeBy &lt;-<span class="st"> "Column data"</span>
cdArgs<span class="op">$</span>ColorByColData &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"Core.Type"</span>, <span class="st">"TOTAL_DUP"</span>)
cdArgs<span class="op">$</span>ShapeByColData &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"Core.Type"</span>, <span class="st">"driver_1_s"</span>)

init &lt;-<span class="st"> </span><span class="kw">DataFrame</span>(
    <span class="dt">Name =</span> <span class="kw">c</span>(<span class="st">"Column data plot 1"</span>, <span class="st">"Column data plot 2"</span>),
    <span class="dt">Width =</span> <span class="kw">c</span>(<span class="dv">6</span>, <span class="dv">6</span>)
)
app &lt;-<span class="st"> </span><span class="kw"><a href="../reference/iSEE.html">iSEE</a></span>(sce, <span class="dt">initialPanels =</span> init, <span class="dt">colDataArgs =</span> cdArgs)</code></pre></div>
<p>Note that points may only be shaped by a categorical variable.</p>
</div>
<div id="configuring-plot-facets" class="section level2">
<h2 class="hasAnchor">
<a href="#configuring-plot-facets" class="anchor"></a>Configuring plot facets</h2>
<p>Categorical variables may also be used to facet plots by row and column.</p>
<p>For instance, <em>column data plot</em> panels can be facet by variables stored in the columns of <code>colData(sce)</code>. Users can enable faceting by setting <code>FacetByRow</code> and/or <code>FacetByColumn</code> columns to <code>TRUE</code>, and set <code>RowFacetByColData</code> and <code>ColumnFacetByColData</code> to the appropriate column name in <code>colData(sce)</code>. We demonstrate below how faceting may be enable by row, column, or both:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cdArgs &lt;-<span class="st"> </span><span class="kw"><a href="../reference/colDataPlotDefaults.html">colDataPlotDefaults</a></span>(sce, <span class="dv">3</span>)
cdArgs<span class="op">$</span>VisualBoxOpen &lt;-<span class="st"> </span><span class="ot">TRUE</span>
cdArgs<span class="op">$</span>VisualChoices &lt;-<span class="st"> </span><span class="kw">list</span>(<span class="st">"Facets"</span>, <span class="st">"Facets"</span>, <span class="st">"Facets"</span>)
cdArgs<span class="op">$</span>FacetByRow &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="ot">TRUE</span>, <span class="ot">FALSE</span>, <span class="ot">TRUE</span>)
cdArgs<span class="op">$</span>FacetByColumn &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="ot">FALSE</span>, <span class="ot">TRUE</span>, <span class="ot">TRUE</span>)
cdArgs<span class="op">$</span>RowFacetByColData &lt;-<span class="st"> "driver_1_s"</span>
cdArgs<span class="op">$</span>ColumnFacetByColData &lt;-<span class="st"> "Core.Type"</span>
init &lt;-<span class="st"> </span><span class="kw">DataFrame</span>(
    <span class="dt">Name =</span> <span class="kw">paste</span>(<span class="st">"Column data plot"</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>),
    <span class="dt">Width =</span> <span class="kw">c</span>(<span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">4</span>)
)
app &lt;-<span class="st"> </span><span class="kw"><a href="../reference/iSEE.html">iSEE</a></span>(sce, <span class="dt">initialPanels =</span> init, <span class="dt">colDataArgs =</span> cdArgs)</code></pre></div>
</div>
</div>
<div id="selection-parameters" class="section level1">
<h1 class="hasAnchor">
<a href="#selection-parameters" class="anchor"></a>Selection parameters</h1>
<div id="shiny-brush" class="section level2">
<h2 class="hasAnchor">
<a href="#shiny-brush" class="anchor"></a>Shiny brush</h2>
<p>The initial state of <em>iSEE</em> applications can be configured all the way down to point selections and links between panels. For instance, in the example below, we preconfigure the <code>SelectByPlot</code> column of a <em>column data plot</em> panel to receive a point selection from a <em>reduced dimension plot</em> panel. This requires an active selection in the <em>reduced dimension plot</em> panel, which is achieved by preconfiguring the <code>BrushData</code> column of the arguments.</p>
<p>The simplest way to obtain a valid input for the <code>BrushData</code> column is to launch an <em>iSEE</em> application, make the desired selection using a Shiny brush, open the <em>iSEE</em> code tracker, and copy paste the relevant point selection data. The result should look as below:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Preconfigure the receiver panel</span>
cdArgs &lt;-<span class="st"> </span><span class="kw"><a href="../reference/colDataPlotDefaults.html">colDataPlotDefaults</a></span>(sce, <span class="dv">1</span>)
cdArgs<span class="op">$</span>SelectBoxOpen &lt;-<span class="st"> </span><span class="ot">TRUE</span>
cdArgs<span class="op">$</span>SelectByPlot &lt;-<span class="st"> "Reduced dimension plot 1"</span>
cdArgs<span class="op">$</span>SelectEffect &lt;-<span class="st"> "Color"</span>
cdArgs<span class="op">$</span>SelectColor &lt;-<span class="st"> "darkgoldenrod1"</span>

<span class="co"># Preconfigure the sender panel, including the point selection</span>
rdArgs &lt;-<span class="st"> </span><span class="kw"><a href="../reference/redDimPlotDefaults.html">redDimPlotDefaults</a></span>(sce, <span class="dv">1</span>)
rdArgs<span class="op">$</span>BrushData &lt;-<span class="st"> </span><span class="kw">list</span>(
        <span class="kw">list</span>(<span class="dt">xmin =</span> <span class="fl">2.8</span>, <span class="dt">xmax =</span> <span class="fl">10.4</span>, <span class="dt">ymin =</span> <span class="fl">0.6</span>, <span class="dt">ymax =</span> <span class="fl">13.2</span>, 
        <span class="dt">mapping =</span> <span class="kw">list</span>(<span class="dt">x =</span> <span class="st">"X"</span>, <span class="dt">y =</span> <span class="st">"Y"</span>), <span class="dt">domain =</span> <span class="kw">list</span>(<span class="dt">left =</span> <span class="op">-</span><span class="fl">14.0</span>, <span class="dt">right =</span> <span class="fl">10.9</span>, 
            <span class="dt">bottom =</span> <span class="op">-</span><span class="fl">12.0</span>, <span class="dt">top =</span> <span class="fl">16.4</span>), <span class="dt">range =</span> <span class="kw">list</span>(<span class="dt">left =</span> <span class="fl">38.7</span>, 
            <span class="dt">right =</span> <span class="fl">541.5</span>, <span class="dt">bottom =</span> <span class="fl">466.0</span>, <span class="dt">top =</span> <span class="fl">23.7</span>), 
        <span class="dt">log =</span> <span class="kw">list</span>(<span class="dt">x =</span> <span class="ot">NULL</span>, <span class="dt">y =</span> <span class="ot">NULL</span>), <span class="dt">direction =</span> <span class="st">"xy"</span>, <span class="dt">brushId =</span> <span class="st">"redDimPlot1_Brush"</span>, 
        <span class="dt">outputId =</span> <span class="st">"redDimPlot1"</span>)
    )

init &lt;-<span class="st"> </span><span class="kw">DataFrame</span>(
    <span class="dt">Name =</span> <span class="kw">c</span>(<span class="st">"Reduced dimension plot 1"</span>, <span class="st">"Column data plot 1"</span>),
    <span class="dt">Width =</span> <span class="kw">c</span>(<span class="dv">6</span>, <span class="dv">6</span>)
)
app &lt;-<span class="st"> </span><span class="kw"><a href="../reference/iSEE.html">iSEE</a></span>(sce, <span class="dt">initialPanels =</span> init, <span class="dt">redDimArgs =</span> rdArgs, <span class="dt">colDataArgs =</span> cdArgs)</code></pre></div>
<p>Note that in the example above, we chose to color selected data points in the receiver panel, by setting the <code>SelectEffect</code> argument to <code>"Color"</code>, and the color of selected data points to <code>"darkgoldenrod1"</code>. Other choices include <code>"Restrict"</code>, to show only the selected data points; and <code>"Transparent"</code> (the default), to increase the transparency of unselected data points.</p>
</div>
<div id="lasso-selection" class="section level2">
<h2 class="hasAnchor">
<a href="#lasso-selection" class="anchor"></a>Lasso selection</h2>
<p>An identical process can be followed to preconfigure a lasso point selection:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Preconfigure the receiver panel</span>
cdArgs &lt;-<span class="st"> </span><span class="kw"><a href="../reference/colDataPlotDefaults.html">colDataPlotDefaults</a></span>(sce, <span class="dv">1</span>)
cdArgs<span class="op">$</span>SelectBoxOpen &lt;-<span class="st"> </span><span class="ot">TRUE</span>
cdArgs<span class="op">$</span>SelectByPlot &lt;-<span class="st"> "Reduced dimension plot 1"</span>
cdArgs<span class="op">$</span>SelectEffect &lt;-<span class="st"> "Color"</span>
cdArgs<span class="op">$</span>SelectColor &lt;-<span class="st"> "darkgoldenrod1"</span>

<span class="co"># Preconfigure the sender panel, including the point selection</span>
rdArgs &lt;-<span class="st"> </span><span class="kw"><a href="../reference/redDimPlotDefaults.html">redDimPlotDefaults</a></span>(sce, <span class="dv">1</span>)
rdArgs<span class="op">$</span>LassoData &lt;-<span class="st"> </span><span class="kw">list</span>(
    <span class="kw">list</span>(<span class="dt">lasso =</span> <span class="ot">NULL</span>, <span class="dt">closed =</span> <span class="ot">TRUE</span>, <span class="dt">panelvar1 =</span> <span class="ot">NULL</span>, <span class="dt">panelvar2 =</span> <span class="ot">NULL</span>, <span class="dt">mapping =</span> <span class="kw">list</span>(
        <span class="dt">x =</span> <span class="st">"X"</span>, <span class="dt">y =</span> <span class="st">"Y"</span>), <span class="dt">coord =</span> <span class="kw">structure</span>(<span class="kw">c</span>(<span class="fl">9.7</span>, <span class="fl">4.0</span>, <span class="fl">2.0</span>, 
        <span class="fl">8.2</span>, <span class="fl">10.5</span>, <span class="fl">9.7</span>, <span class="fl">9.0</span>, <span class="fl">12.8</span>, <span class="fl">7.9</span>, <span class="fl">0.9</span>, <span class="fl">2.1</span>, <span class="fl">9.0</span>), <span class="dt">.Dim =</span> <span class="kw">c</span>(6L, 
        2L)))
    )

init &lt;-<span class="st"> </span><span class="kw">DataFrame</span>(
    <span class="dt">Name =</span> <span class="kw">c</span>(<span class="st">"Reduced dimension plot 1"</span>, <span class="st">"Column data plot 1"</span>),
    <span class="dt">Width =</span> <span class="kw">c</span>(<span class="dv">6</span>, <span class="dv">6</span>)
)
app &lt;-<span class="st"> </span><span class="kw"><a href="../reference/iSEE.html">iSEE</a></span>(sce, <span class="dt">initialPanels =</span> init, <span class="dt">redDimArgs =</span> rdArgs, <span class="dt">colDataArgs =</span> cdArgs)</code></pre></div>
</div>
</div>
<div id="parameter-fields-accepting-both-integercharacter" class="section level1">
<h1 class="hasAnchor">
<a href="#parameter-fields-accepting-both-integercharacter" class="anchor"></a>Parameter fields accepting both integer/character</h1>
<p>As we mentioned <a href="#assay">earlier</a>, certain parameters such as assay types may be set to either character or integer values. Those include parameters referring to assays <code>*Assay</code>, features <code>*FeatName</code>, and samples <code>*Sample</code>.</p>
<p>We generally recommend to set those parameters using integer values; those are used internally and will always work even in the presence of unnamed assays, for instance:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fexArg &lt;-<span class="st"> </span><span class="kw"><a href="../reference/featAssayPlotDefaults.html">featAssayPlotDefaults</a></span>(sce, <span class="dv">2</span>)
fexArg<span class="op">$</span>DataBoxOpen &lt;-<span class="st"> </span><span class="ot">TRUE</span>
fexArg<span class="op">$</span>Assay &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">6</span>, <span class="dv">6</span>)
fexArg<span class="op">$</span>YAxisFeatName &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">10</span>, <span class="dv">11</span>)
fexArg<span class="op">$</span>XAxis &lt;-<span class="st"> "Feature name"</span>
fexArg<span class="op">$</span>XAxisFeatName &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">12</span>, <span class="dv">13</span>)
init &lt;-<span class="st"> </span><span class="kw">DataFrame</span>(
    <span class="dt">Name =</span> <span class="kw">paste</span>(<span class="st">"Feature assay plot"</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">2</span>),
    <span class="dt">Width =</span> <span class="kw">c</span>(<span class="dv">6</span>, <span class="dv">6</span>)
)
app &lt;-<span class="st"> </span><span class="kw"><a href="../reference/iSEE.html">iSEE</a></span>(sce, <span class="dt">initialPanels =</span> init, <span class="dt">featAssayArgs =</span> fexArg)</code></pre></div>
<p>Alternatively, users may refer to assays, features, and samples by their character name, if any. However, users are advised against mixing character and integer values, as this will cause integer values to be converted to the character type.</p>
<p>For instance, the example below will fail to find an assay called <code>"6"</code> for the first panel, and it will instead fall back to using the first assay (<code>"tophat_counts</code>):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fexArg &lt;-<span class="st"> </span><span class="kw"><a href="../reference/featAssayPlotDefaults.html">featAssayPlotDefaults</a></span>(sce, <span class="dv">2</span>)
fexArg<span class="op">$</span>DataBoxOpen &lt;-<span class="st"> </span><span class="ot">TRUE</span>
fexArg<span class="op">$</span>Assay &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">6</span>, <span class="st">"counts"</span>)
init &lt;-<span class="st"> </span><span class="kw">DataFrame</span>(
    <span class="dt">Name =</span> <span class="kw">paste</span>(<span class="st">"Feature assay plot"</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">2</span>),
    <span class="dt">Width =</span> <span class="kw">c</span>(<span class="dv">6</span>, <span class="dv">6</span>)
)
app &lt;-<span class="st"> </span><span class="kw"><a href="../reference/iSEE.html">iSEE</a></span>(sce, <span class="dt">initialPanels =</span> init, <span class="dt">featAssayArgs =</span> fexArg)</code></pre></div>
<p>We provide below an example where all parameters are defined as character values:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fexArg &lt;-<span class="st"> </span><span class="kw"><a href="../reference/featAssayPlotDefaults.html">featAssayPlotDefaults</a></span>(sce, <span class="dv">2</span>)
fexArg<span class="op">$</span>DataBoxOpen &lt;-<span class="st"> </span><span class="ot">TRUE</span>
fexArg<span class="op">$</span>Assay &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"logcounts"</span>, <span class="st">"counts"</span>)
fexArg<span class="op">$</span>YAxisFeatName &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"A3galt2"</span>, <span class="st">"A4galt"</span>)
fexArg<span class="op">$</span>XAxis &lt;-<span class="st"> "Feature name"</span>
fexArg<span class="op">$</span>XAxisFeatName &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"L3mbtl1"</span>, <span class="st">"L3mbtl2"</span>)
init &lt;-<span class="st"> </span><span class="kw">DataFrame</span>(
    <span class="dt">Name =</span> <span class="kw">paste</span>(<span class="st">"Feature assay plot"</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">2</span>),
    <span class="dt">Width =</span> <span class="kw">c</span>(<span class="dv">6</span>, <span class="dv">6</span>)
)
app &lt;-<span class="st"> </span><span class="kw"><a href="../reference/iSEE.html">iSEE</a></span>(sce, <span class="dt">initialPanels =</span> init, <span class="dt">featAssayArgs =</span> fexArg)</code></pre></div>
<p>We refer users to the individual help page of each panel type listed <a href="#further">below</a> to learn more about the parameters of each type of panel that accept integer and character values.</p>
</div>
<div id="writing-your-own-tour" class="section level1">
<h1 class="hasAnchor">
<a href="#writing-your-own-tour" class="anchor"></a>Writing your own tour</h1>
<p>By providing a data frame to the <code>tour</code> argument of <code>iSEE</code>, you can create your own tour that will start when the app is launched<a href="#fn2" class="footnoteRef" id="fnref2"><sup>2</sup></a>. The data frame should have two columns, <code>element</code> and <code>intro</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">introtour &lt;-<span class="st"> </span><span class="kw">read.delim</span>(
    <span class="kw">system.file</span>(<span class="st">"extdata/intro_firststeps.txt"</span>, <span class="dt">package =</span> <span class="st">"iSEE"</span>), 
    <span class="dt">sep =</span> <span class="st">";"</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>)
<span class="kw">head</span>(introtour)
<span class="co">#&gt;           element</span>
<span class="co">#&gt; 1        #Welcome</span>
<span class="co">#&gt; 2      #allpanels</span>
<span class="co">#&gt; 3    #redDimPlot1</span>
<span class="co">#&gt; 4   #colDataPlot1</span>
<span class="co">#&gt; 5 #featAssayPlot1</span>
<span class="co">#&gt; 6  #rowStatTable1</span>
<span class="co">#&gt;                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 intro</span>
<span class="co">#&gt; 1 Welcome to the interactive tour for iSEE - the Interactive SummarizedExperiment Explorer.&lt;br/&gt;&lt;br/&gt;You will be shown around the different components of iSEE and learn the basic usage mechanisms by doing. Highlighted elements will respond to the user's actions, while the rest of the UI will be shaded. You will be prompted at particular spots to perform some actions, which will be marked with &lt;strong&gt;Action:&lt;/strong&gt; text. Please take care to follow these instructions, since later parts of the tour may assume that all the actions from previous steps have been performed.&lt;br/&gt;&lt;br/&gt;&lt;strong&gt;Action:&lt;/strong&gt; now, click on the 'Next' button or use the right arrow of your keyboard to proceed into your tour.</span>
<span class="co">#&gt; 2      iSEE provides a Shiny interface that allows you to generate a series of panels for exploring &lt;code&gt;SummarizedExperiment&lt;/code&gt; objects. Here, we use single-cell RNA sequencing data from the &lt;a href=https://doi.org/doi:10.18129/B9.bioc.scRNAseq&gt;&lt;i&gt;scRNAseq&lt;/i&gt; package&lt;/a&gt;, specifically a subset of gene expression profiles from cells in the mouse visual cortex &lt;a href=https://doi.org/10.1038/nn.4216&gt;(Tasic &lt;i&gt;et al.&lt;/i&gt;, 2016)&lt;/a&gt;. Each column of the &lt;code&gt;SummarizedExperiment&lt;/code&gt; corresponds to a cell, while each row corresponds to a gene.&lt;br/&gt;&lt;br/&gt;Using iSEE, you can generate a variety of different panel types to visualize the data. These are described in more detail in the following steps.</span>
<span class="co">#&gt; 3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  For example, you can construct a &lt;font color=#402ee8&gt;Reduced dimension plot&lt;/font&gt; (e.g., PCA, &lt;i&gt;t&lt;/i&gt;-SNE) to efficiently explore high-dimensional datasets. Here, each point represents a cell.</span>
<span class="co">#&gt; 4                                                                                                                                                                                                                                                                                                                                         You can construct a &lt;font color=#402ee8&gt;Column data plot&lt;/font&gt; involving column-level metadata, where each point represents a column (in this case, a cell) in the &lt;code&gt;SummarizedExperiment&lt;/code&gt; object. Points can be displayed using violin plots stratified by an experimental factor on the x-axis, or as a scatter plot involving continuous variables on both the x- and y-axes.</span>
<span class="co">#&gt; 5                                                                                                                                                                                                                                                                                                                                                                                                                                A &lt;font color=#402ee8&gt;Feature assay plot&lt;/font&gt; displays the assay values (here, gene expression) across all cells. Like the &lt;font color=#402ee8&gt;Column data plot&lt;/font&gt;, points represent cells and can be shown in both violin or scatter plots, depending on the variable selected on the x-axis.</span>
<span class="co">#&gt; 6                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      A &lt;font color=#402ee8&gt;Row statistic table&lt;/font&gt; displays values in the &lt;code&gt;rowData&lt;/code&gt; slot of your input object. They can also be used to efficiently select genes for the other plots.</span></code></pre></div>
<p>Each entry of the <code>element</code> column contains the name of a UI element in the application, prefixed by a hash sign (<code>#</code>). The <code>intro</code> column contains the corresponding text (or basic HTML) that is to be shown at each step.</p>
<p>The simplest way to get started is to copy the <code>intro_firststeps.txt</code> file from the <code>inst/extdata</code> folder and edit it for your specific data set. More customized tours require some knowledge of the names of the UI elements to put in the <code>element</code> column. We recommend one of the following options:</p>
<ul>
<li>If using Firefox, open the <code>Tools</code> menu. Select <code>Web Developer</code>, and in the submenu, select <code>Inspector</code>. This will toggle a toolbar, and you will be able to read out the name of the element of interest when you hover and click on it. If you want to select another element, you might need to re-click on the icon in the upper left corner of the toolbox, <code>Pick an element from the page</code>.</li>
<li>If using Safari, open the <code>Develop</code> menu and select <code>Show Web Inspector</code>. To toggle the selection of elements, you need to click on the crosshair icon in the top part of the toolbox, then again, explore the desired element by clicking or hovering on it.</li>
<li>If using Chrome, from the <code>View</code> menu, select first <code>Developer</code> and then <code>Developer Tools</code>. Click then on the selecting arrow in the top left corner, and similarly to the other browsers, hover or click on the element of interest to obtain its name.</li>
<li>Alternatively, the <a href="https://selectorgadget.com">SelectorGadget</a> browser extension can be used.</li>
</ul>
<p>Most elements can be identified using the above strategies. Selectize widgets are trickier but can be handled with, e.g., <code>#heatMapPlot1_ColData + .selectize-control</code>. Please see the <code>intro_firststeps.txt</code> file in the <code>inst/extdata</code> folder for more of these examples.</p>
<p>Sometimes it is useful to place one step of the tour in the center. To do so, simply put a hash sign before a word which does not link directly to any CSS selector (e.g., as we do for <code>#Welcome</code>) in the corresponding <code>element</code> column.</p>
</div>
<div id="further" class="section level1">
<h1 class="hasAnchor">
<a href="#further" class="anchor"></a>Further reading</h1>
<p>Users should refer to the following help pages for the full list of values that can be specified in <code>iSEE</code>:</p>
<ul>
<li><code><a href="../reference/redDimPlotDefaults.html">?redDimPlotDefaults</a></code></li>
<li><code><a href="../reference/colDataPlotDefaults.html">?colDataPlotDefaults</a></code></li>
<li><code><a href="../reference/featAssayPlotDefaults.html">?featAssayPlotDefaults</a></code></li>
<li><code>?rowDataPlotDefaults,</code></li>
<li><code><a href="../reference/rowStatTableDefaults.html">?rowStatTableDefaults</a></code></li>
<li><code><a href="../reference/sampAssayPlotDefaults.html">?sampAssayPlotDefaults</a></code></li>
<li><code><a href="../reference/heatMapPlotDefaults.html">?heatMapPlotDefaults</a></code></li>
<li><code><a href="../reference/pointDefaults.html">?"iSEE point parameters"</a></code></li>
<li><code><a href="../reference/selectDefaults.html">?"iSEE selection parameters"</a></code></li>
</ul>
<p>Some fairly complex configurations for a variety of data sets can be found at <a href="https://github.com/LTLA/iSEE2018" class="uri">https://github.com/LTLA/iSEE2018</a>. These may serve as useful examples for setting up your own configurations.</p>
</div>
<div id="session-info" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()
<span class="co">#&gt; R version 3.5.0 (2018-04-23)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin15.6.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Sierra 10.12.6</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] parallel  stats4    stats     graphics  grDevices utils     datasets </span>
<span class="co">#&gt; [8] methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] iSEE_1.1.9                  SingleCellExperiment_1.3.10</span>
<span class="co">#&gt;  [3] SummarizedExperiment_1.11.6 DelayedArray_0.7.42        </span>
<span class="co">#&gt;  [5] BiocParallel_1.15.12        matrixStats_0.54.0         </span>
<span class="co">#&gt;  [7] Biobase_2.41.2              GenomicRanges_1.33.14      </span>
<span class="co">#&gt;  [9] GenomeInfoDb_1.17.1         IRanges_2.15.18            </span>
<span class="co">#&gt; [11] S4Vectors_0.19.19           BiocGenerics_0.27.1        </span>
<span class="co">#&gt; [13] BiocStyle_2.9.6            </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] nlme_3.1-137           bitops_1.0-6           fs_1.2.6              </span>
<span class="co">#&gt;  [4] bit64_0.9-7            rprojroot_1.3-2        tools_3.5.0           </span>
<span class="co">#&gt;  [7] backports_1.1.2        R6_2.2.2               DT_0.4                </span>
<span class="co">#&gt; [10] vipor_0.4.5            DBI_1.0.0              lazyeval_0.2.1        </span>
<span class="co">#&gt; [13] mgcv_1.8-24            colorspace_1.3-2       tidyselect_0.2.4      </span>
<span class="co">#&gt; [16] bit_1.1-14             compiler_3.5.0         shinyjs_1.0           </span>
<span class="co">#&gt; [19] xml2_1.2.0             desc_1.2.0             colourpicker_1.0      </span>
<span class="co">#&gt; [22] bookdown_0.7           scales_1.0.0           pkgdown_1.1.0         </span>
<span class="co">#&gt; [25] commonmark_1.5         stringr_1.3.1          digest_0.6.17         </span>
<span class="co">#&gt; [28] rmarkdown_1.10         rentrez_1.2.1          XVector_0.21.3        </span>
<span class="co">#&gt; [31] pkgconfig_2.0.2        htmltools_0.3.6        htmlwidgets_1.2       </span>
<span class="co">#&gt; [34] rlang_0.2.2            rstudioapi_0.7         RSQLite_2.1.1         </span>
<span class="co">#&gt; [37] shiny_1.1.0            bindr_0.1.1            jsonlite_1.5          </span>
<span class="co">#&gt; [40] dplyr_0.7.6            RCurl_1.95-4.11        magrittr_1.5          </span>
<span class="co">#&gt; [43] GenomeInfoDbData_1.1.0 Matrix_1.2-14          Rcpp_0.12.18          </span>
<span class="co">#&gt; [46] munsell_0.5.0          rintrojs_0.2.0         stringi_1.2.4         </span>
<span class="co">#&gt; [49] yaml_2.2.0             MASS_7.3-50            zlibbioc_1.27.0       </span>
<span class="co">#&gt; [52] plyr_1.8.4             grid_3.5.0             blob_1.1.1            </span>
<span class="co">#&gt; [55] promises_1.0.1         shinydashboard_0.7.0   crayon_1.3.4          </span>
<span class="co">#&gt; [58] miniUI_0.1.1.1         lattice_0.20-35        cowplot_0.9.3         </span>
<span class="co">#&gt; [61] knitr_1.20             pillar_1.3.0           igraph_1.2.2          </span>
<span class="co">#&gt; [64] reshape2_1.4.3         XML_3.98-1.16          glue_1.3.0            </span>
<span class="co">#&gt; [67] evaluate_0.11          BiocManager_1.30.2     httpuv_1.4.5          </span>
<span class="co">#&gt; [70] gtable_0.2.0           purrr_0.2.5            assertthat_0.2.0      </span>
<span class="co">#&gt; [73] ggplot2_3.0.0          xfun_0.3               mime_0.5              </span>
<span class="co">#&gt; [76] xtable_1.8-3           roxygen2_6.1.0         later_0.7.5           </span>
<span class="co">#&gt; [79] viridisLite_0.3.0      tibble_1.4.2           AnnotationDbi_1.43.1  </span>
<span class="co">#&gt; [82] memoise_1.1.0          bindrcpp_0.2.2         shinyAce_0.3.1</span>
<span class="co"># devtools::session_info()</span></code></pre></div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-kra2018iSEE">
<p>Rue-Albrecht, K., F. Marini, C. Soneson, and A. T. L. Lun. 2018. “ISEE: Interactive Summarizedexperiment Explorer.” <em>F1000Research</em> 7 (June): 741.</p>
</div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>We’ll re-use the example from the <a href="https://bioconductor.org/packages/3.8/iSEE/vignettes/basic.html">previous workflow</a>.<a href="#fnref1">↩</a></p></li>
<li id="fn2"><p>In theory. On servers, sometimes the tour does not recognize the UI elements at start-up and needs to be restarted via the “Click me for quick tour” button to work properly.<a href="#fnref2">↩</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#changing-the-default-start-configuration">Changing the default start configuration</a></li>
      <li>
<a href="#data-parameters">Data parameters</a><ul class="nav nav-pills nav-stacked">
<li><a href="#overview">Overview</a></li>
      <li><a href="#setting-the-y-axis">Setting the Y-axis</a></li>
      <li><a href="#xaxis">Setting the X-axis</a></li>
      <li><a href="#configuring-the-type-of-dimensionality-reduction">Configuring the type of dimensionality reduction</a></li>
      <li><a href="#assay">Configuring the type of assay data</a></li>
      </ul>
</li>
      <li>
<a href="#visual-parameters">Visual parameters</a><ul class="nav nav-pills nav-stacked">
<li><a href="#overview-1">Overview</a></li>
      <li><a href="#configuring-default-visual-parameters">Configuring default visual parameters</a></li>
      <li><a href="#linking-point-aesthetics-to-variables">Linking point aesthetics to variables</a></li>
      <li><a href="#configuring-plot-facets">Configuring plot facets</a></li>
      </ul>
</li>
      <li>
<a href="#selection-parameters">Selection parameters</a><ul class="nav nav-pills nav-stacked">
<li><a href="#shiny-brush">Shiny brush</a></li>
      <li><a href="#lasso-selection">Lasso selection</a></li>
      </ul>
</li>
      <li><a href="#parameter-fields-accepting-both-integercharacter">Parameter fields accepting both integer/character</a></li>
      <li><a href="#writing-your-own-tour">Writing your own tour</a></li>
      <li><a href="#further">Further reading</a></li>
      <li><a href="#session-info">Session Info</a></li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by <a href="link_to_charlotte">Charlotte Soneson</a>, <a href="link_to_aaron">Aaron Lun</a>, <a href="https://federicomarini.github.io">Federico Marini</a>, <a href="link_to_kevin">Kevin Rue-Albrecht</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  
</body>
</html>
