<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Deploying custom panels in the <code>iSEE</code> interface • iSEE</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Deploying custom panels in the &lt;code&gt;iSEE&lt;/code&gt; interface">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">iSEE</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">1.1.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/basic.html">An introduction to the `r Biocpkg("iSEE")` interface 
</a>
    </li>
    <li>
      <a href="../articles/configure.html">Configuring `r Biocpkg("iSEE")` apps
</a>
    </li>
    <li>
      <a href="../articles/custom.html">Deploying custom panels in the `iSEE` interface</a>
    </li>
    <li>
      <a href="../articles/ecm.html">Describing the `ExperimentColorMap` class</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/csoneson/iSEE">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><link href="custom_files/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet">
<script src="custom_files/font-awesome-5.0.13/js/fontawesome-all.min.js"></script><script src="custom_files/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Deploying custom panels in the <code>iSEE</code> interface</h1>
                        <h4 class="author">Federico Marini</h4>
            <address class="author_afil">
      Institute of Medical Biostatistics, Epidemiology and Informatics (IMBEI), Mainz<br><a class="author_email" href="mailto:#"></a><a href="mailto:marinif@uni-mainz.de">marinif@uni-mainz.de</a>
      </address>
                              <h4 class="author">Aaron Lun</h4>
            <address class="author_afil">
      Cancer Research UK Cambridge Institute, University of Cambridge<br><a class="author_email" href="mailto:#"></a><a href="mailto:aaron.lun@cruk.cam.ac.uk">aaron.lun@cruk.cam.ac.uk</a>
      </address>
                              <h4 class="author">Charlotte Soneson</h4>
            <address class="author_afil">
      Institute of Molecular Life Sciences, University of Zurich<br><a class="author_email" href="mailto:#"></a><a href="mailto:charlottesoneson@gmail.com">charlottesoneson@gmail.com</a>
      </address>
                              <h4 class="author">Kevin Rue-Albrecht</h4>
            <address class="author_afil">
      Kennedy Institute of Rheumatology, University of Oxford, Headington, Oxford OX3 7FY, UK.<br><a class="author_email" href="mailto:#"></a><a href="mailto:kevin.rue-albrecht@kennedy.ox.ac.uk">kevin.rue-albrecht@kennedy.ox.ac.uk</a>
      </address>
                  
            <h4 class="date">25 September 2018</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/csoneson/iSEE/blob/master/vignettes/custom.Rmd"><code>vignettes/custom.Rmd</code></a></small>
      <div class="hidden name"><code>custom.Rmd</code></div>

    </div>

    
    
<p><strong>Compiled date</strong>: 2018-09-25</p>
<p><strong>Last edited</strong>: 2018-03-08</p>
<p><strong>License</strong>: MIT + file LICENSE</p>
<div id="background" class="section level1">
<h1 class="hasAnchor">
<a href="#background" class="anchor"></a>Background</h1>
<p>Users can define their own plots or tables to include in the <code>iSEE</code> interface <span class="citation">(Rue-Albrecht et al. 2018)</span>. These custom panels are intended to receive a subset or rows or columns from other transmitting panels in the interface. The values in the custom panels will be recomputed “on the fly” by user-defined functions with the transmitted subset. This provides a flexible and convenient approach for light-weight interactive analysis during data exploration. For example, selection of a particular subset of samples can be transmitted to a custom plot panel that performs dimensionality reduction on that subset. Alternatively, the subset can be transmitted to a custom table that performs a differential expression analysis between that subset and all other samples.</p>
</div>
<div id="defining-custom-functions" class="section level1">
<h1 class="hasAnchor">
<a href="#defining-custom-functions" class="anchor"></a>Defining custom functions</h1>
<div id="minimum-requirements" class="section level2">
<h2 class="hasAnchor">
<a href="#minimum-requirements" class="anchor"></a>Minimum requirements</h2>
<p>Recalculations in custom panels are performed using user-defined functions that are supplied to the <code><a href="../reference/iSEE.html">iSEE()</a></code> call. The only requirements are that the function has to accept:</p>
<ul>
<li>A <code>SummarizedExperiment</code> object or its derivatives as the first argument.</li>
<li>A vector of row names as the second argument, specifying the rows that are selected in the transmitting <em>row-panel</em> panel.</li>
<li>A vector of column names as the third argument, specifying the columns that are selected in the transmitting <em>column-based</em> panel.</li>
</ul>
<p>In addition:</p>
<ul>
<li>For <em>custom plot panels</em>, the function is expected to produce a <code>ggplot</code> object.</li>
<li>For <em>custom table panels</em>, the function is expected to produce a <code>data.frame</code>.</li>
</ul>
</div>
<div id="example-of-custom-plot-panel" class="section level2">
<h2 class="hasAnchor">
<a href="#example-of-custom-plot-panel" class="anchor"></a>Example of custom plot panel</h2>
<p>To demonstrate the use of <em>custom plot panels</em>, we define an example function <code>CUSTOM_PCA</code> below. This takes a subset of features and cells in a <code>SingleCellExperiment</code> object, performs a principal components analysis (PCA) using that subset with <code>runPCA</code>, and creates a plot of the first two principal components with <code>plotPCA</code> <span class="citation">(McCarthy et al. 2017)</span>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(scater)
CUSTOM_PCA &lt;-<span class="st"> </span><span class="cf">function</span>(se, rows, columns, <span class="dt">colour_by=</span><span class="ot">NULL</span>, <span class="dt">scale_features=</span><span class="ot">TRUE</span>) {
    <span class="cf">if</span> (<span class="op">!</span><span class="kw">is.null</span>(columns)) {
        kept &lt;-<span class="st"> </span>se[, columns]
    } <span class="cf">else</span> {
        <span class="kw">return</span>(
            <span class="kw">ggplot</span>() <span class="op">+</span><span class="st"> </span><span class="kw">theme_void</span>() <span class="op">+</span><span class="st"> </span><span class="kw">geom_text</span>(
                <span class="kw">aes</span>(x, y, <span class="dt">label=</span>label),
                <span class="kw">data.frame</span>(<span class="dt">x=</span><span class="dv">0</span>, <span class="dt">y=</span><span class="dv">0</span>, <span class="dt">label=</span><span class="st">"No column data selected."</span>),
                <span class="dt">size=</span><span class="dv">5</span>)
            )
    }

    scale_features &lt;-<span class="st"> </span><span class="kw">as.logical</span>(scale_features)
    kept &lt;-<span class="st"> </span><span class="kw"><a href="http://www.rdocumentation.org/packages/scater/topics/runPCA">runPCA</a></span>(kept, <span class="dt">feature_set=</span>rows, <span class="dt">scale_features=</span>scale_features)
    <span class="kw"><a href="http://www.rdocumentation.org/packages/scater/topics/plot_reddim">plotPCA</a></span>(kept, <span class="dt">colour_by=</span>colour_by)
}</code></pre></div>
<p>Note that <code>rows</code> and <code>columns</code> may be <code>NULL</code> if no selection was made in the respective transmitting panels. How these should be treated is up to the user-defined function. In the <code>CUSTOM_PCA</code> example above, an empty <em>ggplot</em> is returned if there is no selection on the columns, while the default <code>runPCA</code> behaviour is used if <code>feature_set=NULL</code>.</p>
</div>
<div id="example-of-custom-table-panel" class="section level2">
<h2 class="hasAnchor">
<a href="#example-of-custom-table-panel" class="anchor"></a>Example of custom table panel</h2>
<p>To demonstrate the use of <em>custom table panels</em>, we define an example function <code>CUSTOM_SUMMARY</code> below. This takes a subset of features and cells in a <code>SingleCellExperiment</code> object, and creates data-frame that details the <code>mean</code>, <code>variance</code> and count of samples with expression above a given cut-off within the selection.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">CUSTOM_SUMMARY &lt;-<span class="st"> </span><span class="cf">function</span>(se, ri, ci, <span class="dt">assay=</span><span class="st">"logcounts"</span>, <span class="dt">min_exprs=</span><span class="dv">0</span>) {
    <span class="cf">if</span> (<span class="kw">is.null</span>(ri)) {
        ri &lt;-<span class="st"> </span><span class="kw">rownames</span>(se)
    }
    <span class="cf">if</span> (<span class="kw">is.null</span>(ci)) {
        ci &lt;-<span class="st"> </span><span class="kw">colnames</span>(se)
    }
    
    assayMatrix &lt;-<span class="st"> </span><span class="kw">assay</span>(se, assay)[ri, ci, drop=<span class="ot">FALSE</span>]
    
    <span class="kw">data.frame</span>(
        <span class="dt">Mean =</span> <span class="kw">rowMeans</span>(assayMatrix),
        <span class="dt">Var =</span> <span class="kw">rowVars</span>(assayMatrix),
        <span class="dt">Sum =</span> <span class="kw">rowSums</span>(assayMatrix),
        <span class="dt">n_detected =</span> <span class="kw">rowSums</span>(assayMatrix <span class="op">&gt;</span><span class="st"> </span>min_exprs),
        <span class="dt">row.names =</span> ri
    )
}</code></pre></div>
<p>Note that in the <code>CUSTOM_SUMMARY</code> example above, if either <code>rows</code> or <code>columns</code> are <code>NULL</code>, all rows or columns are used, respectively.</p>
</div>
<div id="additional-arguments" class="section level2">
<h2 class="hasAnchor">
<a href="#additional-arguments" class="anchor"></a>Additional arguments</h2>
<p>Users of custom panels can specify additional arguments via a text box that accepts multi-line inputs. This is automatically converted into named arguments before being passed to the custom function. Each line corresponds to one argument:value pair, separated by the first space (ignoring whitespace at the start of the line).</p>
<p>For example, consider the <code>CUSTOM_PCA</code> function. If a user were to type:</p>
<pre><code>colour_by Nanog
scale_features FALSE</code></pre>
<p>… in the text box, this would pass <code>colour_by="Nanog"</code> and <code>scale_features="TRUE"</code> to <code>CUSTOM_PCA</code>. Note that all additional arguments are passed as character strings for the sake of security. It is the responsibility of the custom function to perform any necessary type checks and conversions (hence the <code>as.logical</code> in <code>CUSTOM_PCA</code>).</p>
<p>Similarly, additional arguments can be passed to the <code>CUSTOM_SUMMARY</code> function above. For instance:</p>
<pre><code>assay counts
min_exprs 5</code></pre>
</div>
<div id="example-use" class="section level2">
<h2 class="hasAnchor">
<a href="#example-use" class="anchor"></a>Example use</h2>
<p>Using the <code>sce</code> object that we generated <a href="https://bioconductor.org/packages/3.8/iSEE/vignettes/basic.html">earlier</a>, we will add some fields for examining the mean-variance relationship across features:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">rowData</span>(sce)<span class="op">$</span>mean_log &lt;-<span class="st"> </span><span class="kw">rowMeans</span>(<span class="kw">logcounts</span>(sce))
<span class="kw">rowData</span>(sce)<span class="op">$</span>var_log &lt;-<span class="st"> </span><span class="kw">apply</span>(<span class="kw">logcounts</span>(sce), <span class="dv">1</span>, var)</code></pre></div>
<p>We will then set up an <code>iSEE</code> instance with four panels - a reduced dimension plot, a row data plot, a custom plot, and a custom table. Note how both custom panels initially receives a column selection from the reduced dimension plot and a row selection from the row data plot. We also set the initial function to the name of <code>CUSTOM_PCA</code> in <code>customDataFun</code>, and the initial arguments to the values described above.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(iSEE)
reddim &lt;-<span class="st"> </span><span class="kw"><a href="../reference/redDimPlotDefaults.html">redDimPlotDefaults</a></span>(sce, <span class="dv">1</span>)
rowdat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/rowDataPlotDefaults.html">rowDataPlotDefaults</a></span>(sce, <span class="dv">1</span>)
rowdat<span class="op">$</span>XAxis &lt;-<span class="st"> "Row data"</span>
rowdat<span class="op">$</span>XAxisRowData &lt;-<span class="st"> "mean_log"</span>
rowdat<span class="op">$</span>YAxis &lt;-<span class="st"> "var_log"</span>

cdp &lt;-<span class="st"> </span><span class="kw"><a href="../reference/customDataPlotDefaults.html">customDataPlotDefaults</a></span>(sce, <span class="dv">1</span>)
cdp<span class="op">$</span>Function &lt;-<span class="st"> "CUSTOM_PCA"</span>
cdp<span class="op">$</span>Arguments &lt;-<span class="st"> "colour_by Nanog</span><span class="ch">\n</span><span class="st">scale_features FALSE"</span>
cdp<span class="op">$</span>ColumnSource &lt;-<span class="st"> "Reduced dimension plot 1"</span>
cdp<span class="op">$</span>RowSource &lt;-<span class="st"> "Row data plot 1"</span>

cst &lt;-<span class="st"> </span><span class="kw"><a href="../reference/customStatTableDefaults.html">customStatTableDefaults</a></span>(sce, <span class="dv">1</span>)
cst<span class="op">$</span>Function &lt;-<span class="st"> "CUSTOM_SUMMARY"</span>
cst<span class="op">$</span>Arguments &lt;-<span class="st"> "assay logcounts</span><span class="ch">\n</span><span class="st">min_exprs 1"</span>
cst<span class="op">$</span>ColumnSource &lt;-<span class="st"> "Reduced dimension plot 1"</span>
cst<span class="op">$</span>RowSource &lt;-<span class="st"> "Row data plot 1"</span>

app &lt;-<span class="st"> </span><span class="kw"><a href="../reference/iSEE.html">iSEE</a></span>(
    sce, <span class="dt">redDimArgs=</span>reddim, <span class="dt">rowDataArgs=</span>rowdat, <span class="dt">customDataArgs=</span>cdp, <span class="dt">customStatArgs=</span>cst,
    <span class="dt">initialPanels=</span><span class="kw">DataFrame</span>(
        <span class="dt">Name=</span><span class="kw">c</span>(
            <span class="st">"Reduced dimension plot 1"</span>, <span class="st">"Row data plot 1"</span>,
            <span class="st">"Custom data plot 1"</span>, <span class="st">"Custom statistics table 1"</span>),
        <span class="dt">Width=</span><span class="kw">c</span>(<span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">12</span>)),
    <span class="dt">customDataFun=</span><span class="kw">list</span>(<span class="dt">CUSTOM_PCA=</span>CUSTOM_PCA),
    <span class="dt">customStatFun=</span><span class="kw">list</span>(<span class="dt">CUSTOM_SUMMARY=</span>CUSTOM_SUMMARY)
    )</code></pre></div>
</div>
</div>
<div id="caching-for-greater-efficiency" class="section level1">
<h1 class="hasAnchor">
<a href="#caching-for-greater-efficiency" class="anchor"></a>Caching for greater efficiency</h1>
<p>True R wizards can take advantage of the pass-by-reference activity of environments to cache results throughout the lifetime of the app. This can be used to avoid repeating time-consuming steps that are not affected by changes in certain parameters. For example, we could cache the output of <code>runPCA</code> to avoid repeating the PCA if only <code>colour_by</code> changes.</p>
<p>We will demonstrate this below using a function that computes log-fold changes for the selected features in one subset of samples compared to the others. For any given column selection, we compute log-fold changes for all features and cache the results in the <code>caching</code> environment. This allows us to avoid recomputing these values if only the row selection changes.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">caching &lt;-<span class="st"> </span><span class="kw">new.env</span>()
CUSTOM_LFC &lt;-<span class="st"> </span><span class="cf">function</span>(se, rows, columns) {
    <span class="cf">if</span> (<span class="kw">is.null</span>(columns)) {
        <span class="kw">return</span>(<span class="kw">data.frame</span>(<span class="dt">logFC=</span><span class="kw">numeric</span>(<span class="dv">0</span>)))
    }

    <span class="cf">if</span> (<span class="op">!</span><span class="kw">identical</span>(caching<span class="op">$</span>columns, columns)) {
        caching<span class="op">$</span>columns &lt;-<span class="st"> </span>columns
        in.subset &lt;-<span class="st"> </span><span class="kw">rowMeans</span>(<span class="kw">logcounts</span>(sce)[,columns])
        out.subset &lt;-<span class="st"> </span><span class="kw">rowMeans</span>(<span class="kw">logcounts</span>(sce)[,<span class="kw">setdiff</span>(<span class="kw">colnames</span>(sce), columns)])
        caching<span class="op">$</span>logFC &lt;-<span class="st"> </span><span class="kw">setNames</span>(in.subset <span class="op">-</span><span class="st"> </span>out.subset, <span class="kw">rownames</span>(sce))
    } 
       
    lfc &lt;-<span class="st"> </span>caching<span class="op">$</span>logFC
    <span class="cf">if</span> (<span class="op">!</span><span class="kw">is.null</span>(rows)) {
        out &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">logFC=</span>lfc[rows], <span class="dt">row.names=</span>rows)
    } <span class="cf">else</span> {
        out &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">logFC=</span>lfc, <span class="dt">row.names=</span><span class="kw">rownames</span>(se))
    }
    out
}</code></pre></div>
<p>We will then set up an <code>iSEE</code> instance with three panels - a reduced dimension plot, a row data plot and a custom statistics table. This is much the same as described for the custom plot panel.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cst &lt;-<span class="st"> </span><span class="kw"><a href="../reference/customStatTableDefaults.html">customStatTableDefaults</a></span>(sce, <span class="dv">1</span>)
cst<span class="op">$</span>Function &lt;-<span class="st"> "CUSTOM_LFC"</span>
cst<span class="op">$</span>ColumnSource &lt;-<span class="st"> "Reduced dimension plot 1"</span>
cst<span class="op">$</span>RowSource &lt;-<span class="st"> "Row data plot 1"</span>

app2 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/iSEE.html">iSEE</a></span>(sce, <span class="dt">redDimArgs=</span>reddim, <span class="dt">rowDataArgs=</span>rowdat, <span class="dt">customStatArgs=</span>cst, 
    <span class="dt">initialPanels=</span><span class="kw">DataFrame</span>(<span class="dt">Name=</span><span class="kw">c</span>(<span class="st">"Reduced dimension plot 1"</span>, 
            <span class="st">"Row data plot 1"</span>, <span class="st">"Custom statistics table 1"</span>)),
    <span class="dt">customStatFun=</span><span class="kw">list</span>(<span class="dt">CUSTOM_LFC=</span>CUSTOM_LFC))</code></pre></div>
</div>
<div id="session-info" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()
<span class="co">#&gt; R version 3.5.0 (2018-04-23)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin15.6.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Sierra 10.12.6</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] parallel  stats4    stats     graphics  grDevices utils     datasets </span>
<span class="co">#&gt; [8] methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] iSEE_1.1.9                  scater_1.9.20              </span>
<span class="co">#&gt;  [3] ggplot2_3.0.0               SingleCellExperiment_1.3.10</span>
<span class="co">#&gt;  [5] SummarizedExperiment_1.11.6 DelayedArray_0.7.42        </span>
<span class="co">#&gt;  [7] BiocParallel_1.15.12        matrixStats_0.54.0         </span>
<span class="co">#&gt;  [9] Biobase_2.41.2              GenomicRanges_1.33.14      </span>
<span class="co">#&gt; [11] GenomeInfoDb_1.17.1         IRanges_2.15.18            </span>
<span class="co">#&gt; [13] S4Vectors_0.19.19           BiocGenerics_0.27.1        </span>
<span class="co">#&gt; [15] BiocStyle_2.9.6            </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] nlme_3.1-137             bitops_1.0-6            </span>
<span class="co">#&gt;  [3] fs_1.2.6                 bit64_0.9-7             </span>
<span class="co">#&gt;  [5] rprojroot_1.3-2          tools_3.5.0             </span>
<span class="co">#&gt;  [7] backports_1.1.2          R6_2.2.2                </span>
<span class="co">#&gt;  [9] DT_0.4                   HDF5Array_1.9.19        </span>
<span class="co">#&gt; [11] vipor_0.4.5              mgcv_1.8-24             </span>
<span class="co">#&gt; [13] DBI_1.0.0                lazyeval_0.2.1          </span>
<span class="co">#&gt; [15] colorspace_1.3-2         withr_2.1.2             </span>
<span class="co">#&gt; [17] tidyselect_0.2.4         gridExtra_2.3           </span>
<span class="co">#&gt; [19] bit_1.1-14               compiler_3.5.0          </span>
<span class="co">#&gt; [21] shinyjs_1.0              xml2_1.2.0              </span>
<span class="co">#&gt; [23] desc_1.2.0               colourpicker_1.0        </span>
<span class="co">#&gt; [25] bookdown_0.7             scales_1.0.0            </span>
<span class="co">#&gt; [27] pkgdown_1.1.0            commonmark_1.5          </span>
<span class="co">#&gt; [29] stringr_1.3.1            digest_0.6.17           </span>
<span class="co">#&gt; [31] rmarkdown_1.10           rentrez_1.2.1           </span>
<span class="co">#&gt; [33] XVector_0.21.3           pkgconfig_2.0.2         </span>
<span class="co">#&gt; [35] htmltools_0.3.6          htmlwidgets_1.2         </span>
<span class="co">#&gt; [37] rlang_0.2.2              rstudioapi_0.7          </span>
<span class="co">#&gt; [39] RSQLite_2.1.1            shiny_1.1.0             </span>
<span class="co">#&gt; [41] DelayedMatrixStats_1.3.9 bindr_0.1.1             </span>
<span class="co">#&gt; [43] jsonlite_1.5             dplyr_0.7.6             </span>
<span class="co">#&gt; [45] RCurl_1.95-4.11          magrittr_1.5            </span>
<span class="co">#&gt; [47] GenomeInfoDbData_1.1.0   Matrix_1.2-14           </span>
<span class="co">#&gt; [49] Rcpp_0.12.18             ggbeeswarm_0.6.0        </span>
<span class="co">#&gt; [51] munsell_0.5.0            Rhdf5lib_1.3.3          </span>
<span class="co">#&gt; [53] viridis_0.5.1            rintrojs_0.2.0          </span>
<span class="co">#&gt; [55] stringi_1.2.4            yaml_2.2.0              </span>
<span class="co">#&gt; [57] MASS_7.3-50              zlibbioc_1.27.0         </span>
<span class="co">#&gt; [59] rhdf5_2.25.11            plyr_1.8.4              </span>
<span class="co">#&gt; [61] grid_3.5.0               blob_1.1.1              </span>
<span class="co">#&gt; [63] promises_1.0.1           shinydashboard_0.7.0    </span>
<span class="co">#&gt; [65] crayon_1.3.4             miniUI_0.1.1.1          </span>
<span class="co">#&gt; [67] lattice_0.20-35          cowplot_0.9.3           </span>
<span class="co">#&gt; [69] knitr_1.20               pillar_1.3.0            </span>
<span class="co">#&gt; [71] igraph_1.2.2             reshape2_1.4.3          </span>
<span class="co">#&gt; [73] XML_3.98-1.16            glue_1.3.0              </span>
<span class="co">#&gt; [75] evaluate_0.11            BiocManager_1.30.2      </span>
<span class="co">#&gt; [77] httpuv_1.4.5             gtable_0.2.0            </span>
<span class="co">#&gt; [79] purrr_0.2.5              assertthat_0.2.0        </span>
<span class="co">#&gt; [81] xfun_0.3                 mime_0.5                </span>
<span class="co">#&gt; [83] xtable_1.8-3             roxygen2_6.1.0          </span>
<span class="co">#&gt; [85] later_0.7.5              viridisLite_0.3.0       </span>
<span class="co">#&gt; [87] tibble_1.4.2             AnnotationDbi_1.43.1    </span>
<span class="co">#&gt; [89] beeswarm_0.2.3           memoise_1.1.0           </span>
<span class="co">#&gt; [91] bindrcpp_0.2.2           shinyAce_0.3.1</span>
<span class="co"># devtools::session_info()</span></code></pre></div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-mccarthy2017scater">
<p>McCarthy, D. J., K. R. Campbell, A. T. Lun, and Q. F. Wills. 2017. “Scater: pre-processing, quality control, normalization and visualization of single-cell RNA-seq data in R.” <em>Bioinformatics</em> 33 (8): 1179–86.</p>
</div>
<div id="ref-kra2018iSEE">
<p>Rue-Albrecht, K., F. Marini, C. Soneson, and A. T. L. Lun. 2018. “ISEE: Interactive Summarizedexperiment Explorer.” <em>F1000Research</em> 7 (June): 741.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#background">Background</a></li>
      <li>
<a href="#defining-custom-functions">Defining custom functions</a><ul class="nav nav-pills nav-stacked">
<li><a href="#minimum-requirements">Minimum requirements</a></li>
      <li><a href="#example-of-custom-plot-panel">Example of custom plot panel</a></li>
      <li><a href="#example-of-custom-table-panel">Example of custom table panel</a></li>
      <li><a href="#additional-arguments">Additional arguments</a></li>
      <li><a href="#example-use">Example use</a></li>
      </ul>
</li>
      <li><a href="#caching-for-greater-efficiency">Caching for greater efficiency</a></li>
      <li><a href="#session-info">Session Info</a></li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by <a href="link_to_charlotte">Charlotte Soneson</a>, <a href="link_to_aaron">Aaron Lun</a>, <a href="https://federicomarini.github.io">Federico Marini</a>, <a href="link_to_kevin">Kevin Rue-Albrecht</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
