<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Describing the <code>ExperimentColorMap</code> class • iSEE</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Describing the &lt;code&gt;ExperimentColorMap&lt;/code&gt; class">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">iSEE</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">1.1.9</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/basic.html">An introduction to the `r Biocpkg("iSEE")` interface 
</a>
    </li>
    <li>
      <a href="../articles/configure.html">Configuring `r Biocpkg("iSEE")` apps
</a>
    </li>
    <li>
      <a href="../articles/custom.html">Deploying custom panels in the `iSEE` interface</a>
    </li>
    <li>
      <a href="../articles/ecm.html">Describing the `ExperimentColorMap` class</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/csoneson/iSEE">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><link href="ecm_files/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet">
<script src="ecm_files/font-awesome-5.0.13/js/fontawesome-all.min.js"></script><script src="ecm_files/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Describing the <code>ExperimentColorMap</code> class</h1>
                        <h4 class="author">Federico Marini</h4>
            <address class="author_afil">
      Institute of Medical Biostatistics, Epidemiology and Informatics (IMBEI), Mainz<br><a class="author_email" href="mailto:#"></a><a href="mailto:marinif@uni-mainz.de">marinif@uni-mainz.de</a>
      </address>
                              <h4 class="author">Aaron Lun</h4>
            <address class="author_afil">
      Cancer Research UK Cambridge Institute, University of Cambridge<br><a class="author_email" href="mailto:#"></a><a href="mailto:aaron.lun@cruk.cam.ac.uk">aaron.lun@cruk.cam.ac.uk</a>
      </address>
                              <h4 class="author">Charlotte Soneson</h4>
            <address class="author_afil">
      Institute of Molecular Life Sciences, University of Zurich<br><a class="author_email" href="mailto:#"></a><a href="mailto:charlottesoneson@gmail.com">charlottesoneson@gmail.com</a>
      </address>
                              <h4 class="author">Kevin Rue-Albrecht</h4>
            <address class="author_afil">
      Kennedy Institute of Rheumatology, University of Oxford, Headington, Oxford OX3 7FY, UK.<br><a class="author_email" href="mailto:#"></a><a href="mailto:kevin.rue-albrecht@kennedy.ox.ac.uk">kevin.rue-albrecht@kennedy.ox.ac.uk</a>
      </address>
                  
            <h4 class="date">25 September 2018</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/csoneson/iSEE/blob/master/vignettes/ecm.Rmd"><code>vignettes/ecm.Rmd</code></a></small>
      <div class="hidden name"><code>ecm.Rmd</code></div>

    </div>

    
    
<p><strong>Compiled date</strong>: 2018-09-25</p>
<p><strong>Last edited</strong>: 2018-03-08</p>
<p><strong>License</strong>: MIT + file LICENSE</p>
<div id="background" class="section level1">
<h1 class="hasAnchor">
<a href="#background" class="anchor"></a>Background</h1>
<p><em><a href="https://bioconductor.org/packages/3.8/iSEE">iSEE</a></em> coordinates the coloration in every plot via the <code>ExperimentColorMap</code> class <span class="citation">(Rue-Albrecht et al. 2018)</span>. Colors for samples or features are defined from column or row metadata or assay values using “colormaps”. Each colormap is a function that takes a single integer argument and returns that number of distinct colors. The <code>ExperimentColorMap</code> is a container that stores these functions for use within the <code><a href="../reference/iSEE.html">iSEE()</a></code> function. Users can define their own colormaps to customize coloration for specific assays or covariates.</p>
</div>
<div id="defining-colormaps" class="section level1">
<h1 class="hasAnchor">
<a href="#defining-colormaps" class="anchor"></a>Defining colormaps</h1>
<div id="colormaps-for-continuous-variables" class="section level2">
<h2 class="hasAnchor">
<a href="#colormaps-for-continuous-variables" class="anchor"></a>Colormaps for continuous variables</h2>
<p>For continuous variables, the function will be asked to generate a number of colors (21, by default). Interpolation will then be performed internally to generate a color gradient. Users can use existing color scales like <code><a href="http://www.rdocumentation.org/packages/viridis/topics/reexports">viridis::viridis</a></code> or <code>heat.colors</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Coloring for log-counts:</span>
logcounts_color_fun &lt;-<span class="st"> </span>viridis<span class="op">::</span>viridis</code></pre></div>
<p>It is also possible to use a function that completely ignores any arguments, and simply returns a fixed number of interpolation points:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Coloring for FPKMs:</span>
fpkm_color_fun &lt;-<span class="st"> </span><span class="cf">function</span>(n){
    <span class="kw">c</span>(<span class="st">"black"</span>,<span class="st">"brown"</span>,<span class="st">"red"</span>,<span class="st">"orange"</span>,<span class="st">"yellow"</span>)
}</code></pre></div>
</div>
<div id="colormaps-for-categorical-variables" class="section level2">
<h2 class="hasAnchor">
<a href="#colormaps-for-categorical-variables" class="anchor"></a>Colormaps for categorical variables</h2>
<p>For categorical variables, the function should accept the number of levels and return a color per level. Colors are automatically assigned to factor levels in the specified order of the levels.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Coloring for the 'driver' metadata variable.</span>
driver_color_fun &lt;-<span class="st"> </span><span class="cf">function</span>(n){
    RColorBrewer<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/RColorBrewer/topics/ColorBrewer">brewer.pal</a></span>(n, <span class="st">"Set2"</span>)
}</code></pre></div>
<p>Alternatively, the function can ignore its arguments and simply return a named vector of colors if users want to specify the color for each level explicitly It is the user’s responsibility to ensure that all levels are accounted for<a href="#fn1" class="footnoteRef" id="fnref1"><sup>1</sup></a>. For instance, the following colormap function will only be compatible with factors of two levels, namely <code>"Y"</code> and <code>"N"</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Coloring for the QC metadata variable:</span>
qc_color_fun &lt;-<span class="st"> </span><span class="cf">function</span>(n){
    qc_colors &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"forestgreen"</span>, <span class="st">"firebrick1"</span>)
    <span class="kw">names</span>(qc_colors) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"Y"</span>, <span class="st">"N"</span>)
    qc_colors
}</code></pre></div>
</div>
</div>
<div id="the-colormap-hierarchy" class="section level1">
<h1 class="hasAnchor">
<a href="#the-colormap-hierarchy" class="anchor"></a>The colormap hierarchy</h1>
<div id="specific-and-shared-colormaps" class="section level2">
<h2 class="hasAnchor">
<a href="#specific-and-shared-colormaps" class="anchor"></a>Specific and shared colormaps</h2>
<p>Colormaps can be defined by users at three different levels:</p>
<ul>
<li>Each individual assay, column data field, and row data field can be assigned its own distinct colormap. Those colormaps are stored as named lists of functions in the <code>assays</code>, <code>colData</code>, and <code>rowData</code> slots, respectively, of the <code>ExperimentColorMap</code>. This can be useful to easily remember which assay is currently shown; to apply different color scale limits to assays that vary on different ranges of values; or display boolean information in an intuitive way, among many other scenarios.</li>
<li>
<em>Shared</em> colormaps can be defined for all assays, all column data, and all row data. These colormaps are stored in the <code>all_discrete</code> and <code>all_continuous</code> slots of the <code>ExperimentColorMap</code>, as lists of functions named <code>assays</code>, <code>colData</code>, and <code>rowData</code>.</li>
<li>
<em>Global</em> colormaps can be defined for all categorical or continuous data. Those two colormaps are stored in the <code>global_discrete</code> and <code>global_continuous</code> slots of the <code>ExperimentColorMap</code>.</li>
</ul>
</div>
<div id="searching-for-colors" class="section level2">
<h2 class="hasAnchor">
<a href="#searching-for-colors" class="anchor"></a>Searching for colors</h2>
<p>When queried for a specific colormap of any type (assay, column data, or row data), the following process takes place:</p>
<ul>
<li>A specific <em>individual</em> colormap is looked up in the appropriate slot of the <code>ExperimentColorMap</code>.</li>
<li>If it is not found, the <em>shared</em> colormap of the appropriate slot is looked up, according to whether the data are categorical or continuous.</li>
<li>If it is not found, the <em>global</em> colormap is looked up, according to whether the data are categorical or continuous.</li>
<li>If none of the above colormaps were defined, the <code>ExperimentColorMap</code> will revert to the default colormaps.</li>
</ul>
<p>By default, <code>viridis</code> is used as the default continuous colormap, and <code>hcl</code> is used as the default categorical colormap.</p>
</div>
</div>
<div id="creating-the-experimentcolormap" class="section level1">
<h1 class="hasAnchor">
<a href="#creating-the-experimentcolormap" class="anchor"></a>Creating the <code>ExperimentColorMap</code>
</h1>
<p>We store the set of colormap functions in an instance of the <code>ExperimentColorMap</code> class. Named functions passed as <code>assays</code>, <code>colData</code>, or <code>rowData</code> arguments will be used for coloring data in those slots, respectively.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(iSEE)
ecm &lt;-<span class="st"> </span><span class="kw"><a href="../reference/ExperimentColorMap-class.html">ExperimentColorMap</a></span>(
    <span class="dt">assays =</span> <span class="kw">list</span>(
        <span class="dt">counts =</span> heat.colors,
        <span class="dt">logcounts =</span> logcounts_color_fun,
        <span class="dt">cufflinks_fpkm =</span> fpkm_color_fun
    ),
    <span class="dt">colData =</span> <span class="kw">list</span>(
        <span class="dt">passes_qc_checks_s =</span> qc_color_fun,
        <span class="dt">driver_1_s =</span> driver_color_fun
    ),
    <span class="dt">all_continuous =</span> <span class="kw">list</span>(
        <span class="dt">assays =</span> viridis<span class="op">::</span>plasma
    )
)
ecm
<span class="co">#&gt; Class: ExperimentColorMap</span>
<span class="co">#&gt; assays(3): counts logcounts cufflinks_fpkm</span>
<span class="co">#&gt; colData(2): passes_qc_checks_s driver_1_s</span>
<span class="co">#&gt; rowData(0):</span>
<span class="co">#&gt; all_discrete(0):</span>
<span class="co">#&gt; all_continuous(1): assays</span></code></pre></div>
<p>Users can change the defaults for all assays or column data by modifying the <em>shared</em> colormaps. Similarly, users can modify the defaults for all continuous or categorical data by modifying the global colormaps. This is demonstrated below for the continuous variables:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/ExperimentColorMap-class.html">ExperimentColorMap</a></span>(
    <span class="dt">all_continuous=</span><span class="kw">list</span>( <span class="co"># shared</span>
        <span class="dt">assays=</span>viridis<span class="op">::</span>plasma,
        <span class="dt">colData=</span>viridis<span class="op">::</span>inferno
    ),
    <span class="dt">global_continuous=</span>viridis<span class="op">::</span>magma <span class="co"># global</span>
)
<span class="co">#&gt; Class: ExperimentColorMap</span>
<span class="co">#&gt; assays(0):</span>
<span class="co">#&gt; colData(0):</span>
<span class="co">#&gt; rowData(0):</span>
<span class="co">#&gt; all_discrete(0):</span>
<span class="co">#&gt; all_continuous(2): assays colData</span>
<span class="co">#&gt; global_continuous(1)</span></code></pre></div>
</div>
<div id="benefits" class="section level1">
<h1 class="hasAnchor">
<a href="#benefits" class="anchor"></a>Benefits</h1>
<p>The <code>ExperimentColorMap</code> class offers the following major features:</p>
<ul>
<li>a single place to define flexible and lightweight sets of colormaps, that may be saved and reused across sessions and projects outside the app, to apply consistent coloring schemes across entire projects</li>
<li>a simple interface through accessors <code><a href="../reference/ExperimentColorMap-class.html">colDataColorMap(colormap, "coldata_name")</a></code> and setters <code>assayColorMap(colormap, "assay_name") &lt;- colormap_function</code>
</li>
<li>an elegant fallback mechanism to consistently return a colormap, even for undefined covariates, including a default categorical and continuous colormap, respectively.</li>
<li>three levels of colormaps override: individual, shared within slot (i.e., assays, colData, rowData), or shared globally between all categorical or continuous data scales.</li>
</ul>
<p>Detailed examples on the use of <code>ExperimentColorMap</code> objects are available in the documentation <code><a href="../reference/ExperimentColorMap-class.html">?ExperimentColorMap</a></code>, as well as below.</p>
</div>
<div id="demonstration" class="section level1">
<h1 class="hasAnchor">
<a href="#demonstration" class="anchor"></a>Demonstration</h1>
<p>We will re-use the <code>allen</code> single-cell RNA-seq data set to demonstrate the use of the <code>ExperimentColorMap</code> class. Using the <code>sce</code> object that we created <a href="https://bioconductor.org/packages/3.8/iSEE/vignettes/basic.html">previously</a>, we create an <code>iSEE</code> app with the <code>SingleCellExperiment</code> object and the colormap generated above.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">app &lt;-<span class="st"> </span><span class="kw"><a href="../reference/iSEE.html">iSEE</a></span>(sce, <span class="dt">colormap =</span> ecm)</code></pre></div>
<p>We run this using <code>runApp</code> to open the app on our browser.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">shiny<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/shiny/topics/runApp">runApp</a></span>(app)</code></pre></div>
<p>Now, choose to color cells by <code>Column data</code> and select <code>passes_qc_checks_s</code>. We will see that all cells that passed QC (<code>Y</code>) are colored “forestgreen”, while the ones that didn’t pass are colored firebrick.</p>
<p>If we color any plot by gene expression, we see that use of counts follows the <code>heat.colors</code> coloring scheme; use of log-counts follows the <code>viridis</code> coloring scheme; and use of FPKMs follows the black-to-yellow scheme we defined in <code>fpkm_color_fun</code>.</p>
</div>
<div id="session-info" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sessionInfo</span>()
<span class="co">#&gt; R version 3.5.0 (2018-04-23)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin15.6.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Sierra 10.12.6</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] parallel  stats4    stats     graphics  grDevices utils     datasets </span>
<span class="co">#&gt; [8] methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] iSEE_1.1.9                  SingleCellExperiment_1.3.10</span>
<span class="co">#&gt;  [3] SummarizedExperiment_1.11.6 DelayedArray_0.7.42        </span>
<span class="co">#&gt;  [5] BiocParallel_1.15.12        matrixStats_0.54.0         </span>
<span class="co">#&gt;  [7] Biobase_2.41.2              GenomicRanges_1.33.14      </span>
<span class="co">#&gt;  [9] GenomeInfoDb_1.17.1         IRanges_2.15.18            </span>
<span class="co">#&gt; [11] S4Vectors_0.19.19           BiocGenerics_0.27.1        </span>
<span class="co">#&gt; [13] BiocStyle_2.9.6            </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] nlme_3.1-137           bitops_1.0-6           fs_1.2.6              </span>
<span class="co">#&gt;  [4] bit64_0.9-7            rprojroot_1.3-2        tools_3.5.0           </span>
<span class="co">#&gt;  [7] backports_1.1.2        R6_2.2.2               DT_0.4                </span>
<span class="co">#&gt; [10] vipor_0.4.5            DBI_1.0.0              lazyeval_0.2.1        </span>
<span class="co">#&gt; [13] mgcv_1.8-24            colorspace_1.3-2       tidyselect_0.2.4      </span>
<span class="co">#&gt; [16] gridExtra_2.3          bit_1.1-14             compiler_3.5.0        </span>
<span class="co">#&gt; [19] shinyjs_1.0            xml2_1.2.0             desc_1.2.0            </span>
<span class="co">#&gt; [22] colourpicker_1.0       bookdown_0.7           scales_1.0.0          </span>
<span class="co">#&gt; [25] pkgdown_1.1.0          commonmark_1.5         stringr_1.3.1         </span>
<span class="co">#&gt; [28] digest_0.6.17          rmarkdown_1.10         rentrez_1.2.1         </span>
<span class="co">#&gt; [31] XVector_0.21.3         pkgconfig_2.0.2        htmltools_0.3.6       </span>
<span class="co">#&gt; [34] htmlwidgets_1.2        rlang_0.2.2            rstudioapi_0.7        </span>
<span class="co">#&gt; [37] RSQLite_2.1.1          shiny_1.1.0            bindr_0.1.1           </span>
<span class="co">#&gt; [40] jsonlite_1.5           dplyr_0.7.6            RCurl_1.95-4.11       </span>
<span class="co">#&gt; [43] magrittr_1.5           GenomeInfoDbData_1.1.0 Matrix_1.2-14         </span>
<span class="co">#&gt; [46] Rcpp_0.12.18           munsell_0.5.0          viridis_0.5.1         </span>
<span class="co">#&gt; [49] rintrojs_0.2.0         stringi_1.2.4          yaml_2.2.0            </span>
<span class="co">#&gt; [52] MASS_7.3-50            zlibbioc_1.27.0        plyr_1.8.4            </span>
<span class="co">#&gt; [55] grid_3.5.0             blob_1.1.1             promises_1.0.1        </span>
<span class="co">#&gt; [58] shinydashboard_0.7.0   crayon_1.3.4           miniUI_0.1.1.1        </span>
<span class="co">#&gt; [61] lattice_0.20-35        cowplot_0.9.3          knitr_1.20            </span>
<span class="co">#&gt; [64] pillar_1.3.0           igraph_1.2.2           reshape2_1.4.3        </span>
<span class="co">#&gt; [67] XML_3.98-1.16          glue_1.3.0             evaluate_0.11         </span>
<span class="co">#&gt; [70] BiocManager_1.30.2     httpuv_1.4.5           gtable_0.2.0          </span>
<span class="co">#&gt; [73] purrr_0.2.5            assertthat_0.2.0       ggplot2_3.0.0         </span>
<span class="co">#&gt; [76] xfun_0.3               mime_0.5               xtable_1.8-3          </span>
<span class="co">#&gt; [79] roxygen2_6.1.0         later_0.7.5            viridisLite_0.3.0     </span>
<span class="co">#&gt; [82] tibble_1.4.2           AnnotationDbi_1.43.1   memoise_1.1.0         </span>
<span class="co">#&gt; [85] bindrcpp_0.2.2         shinyAce_0.3.1</span>
<span class="co"># devtools::session_info()</span></code></pre></div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-kra2018iSEE">
<p>Rue-Albrecht, K., F. Marini, C. Soneson, and A. T. L. Lun. 2018. “ISEE: Interactive Summarizedexperiment Explorer.” <em>F1000Research</em> 7 (June): 741.</p>
</div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>Needless to say, these functions should not be used as shared or global colormaps.<a href="#fnref1">↩</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#background">Background</a></li>
      <li>
<a href="#defining-colormaps">Defining colormaps</a><ul class="nav nav-pills nav-stacked">
<li><a href="#colormaps-for-continuous-variables">Colormaps for continuous variables</a></li>
      <li><a href="#colormaps-for-categorical-variables">Colormaps for categorical variables</a></li>
      </ul>
</li>
      <li>
<a href="#the-colormap-hierarchy">The colormap hierarchy</a><ul class="nav nav-pills nav-stacked">
<li><a href="#specific-and-shared-colormaps">Specific and shared colormaps</a></li>
      <li><a href="#searching-for-colors">Searching for colors</a></li>
      </ul>
</li>
      <li><a href="#creating-the-experimentcolormap">Creating the <code>ExperimentColorMap</code></a></li>
      <li><a href="#benefits">Benefits</a></li>
      <li><a href="#demonstration">Demonstration</a></li>
      <li><a href="#session-info">Session Info</a></li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by <a href="link_to_charlotte">Charlotte Soneson</a>, <a href="link_to_aaron">Aaron Lun</a>, <a href="https://federicomarini.github.io">Federico Marini</a>, <a href="link_to_kevin">Kevin Rue-Albrecht</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
